Solaris grep doesn't know -F.
--- a/tests/deleg-2.tst
+++ b/tests/deleg-2.tst
@@ -18,7 +18,7 @@
 . $SRCDIR/defs || exit 1
 
 # a2ps should fail, saying it does not know the sheet test_delegation.ssh
-$CHK $OPT $TST_FILE -P void 2>&1 | grep -F 'cannot find style sheet' > /dev/null 2>&1
+$CHK $OPT $TST_FILE -P void 2>&1 | ggrep -F 'cannot find style sheet' > /dev/null 2>&1
 
 # Return status of grep
 exit $?

Solaris rm cannot remove current working directory.
--- a/tests/styles.tst
+++ b/tests/styles.tst
@@ -110,6 +110,7 @@
   cat $builddir/sum-ps-diff
   failure=1
 else
+  cd -
   rm -rf $builddir/sum-ps-diff $DIFF
 fi
 
@@ -121,6 +122,7 @@
   cat $builddir/sum-gps-diff
   failure=1
 else
+  cd -
   rm -rf $builddir/sum-gps-diff $GDIFF
 fi
 
This is a fix for:
38408752 Fix a2ps-lpr-wrapper to properly detect /usr/bin/lp
--- a/src/a2ps-lpr-wrapper
+++ b/src/a2ps-lpr-wrapper
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 #
 # a2ps-lpr-wrapper - lp/lpr wrapper script for GNU a2ps
 
