--- tcsh-6.24.06/sh.c.orig	2022-12-19 17:45:42.434084082 +0100
+++ tcsh-6.24.06/sh.c	2022-12-19 17:46:14.302438309 +0100
@@ -173,17 +173,10 @@
     size_t len;
     int add_LOC = 1;
     int add_lang = 1;
-    char trypath[MAXPATHLEN];
-    struct stat st;
 
     if (path == NULL)
         return;
 
-    (void) xsnprintf(trypath, sizeof(trypath), "%s/C/LC_MESSAGES/tcsh.cat",
-	path);
-    if (stat(trypath, &st) == -1)
-	return;
-
     if ((old = getenv("NLSPATH")) != NULL)
         len = strlen(old) + 1;	/* don't forget the colon. */
     else
--- tcsh-6.24.06/sh.func.c.orig	2022-12-19 17:46:32.355990824 +0100
+++ tcsh-6.24.06/sh.func.c	2022-12-19 17:49:15.719213636 +0100
@@ -2625,6 +2625,8 @@
 
     if (adrof(STRcatalog) != NULL)
 	catalog = xasprintf("tcsh.%s", short2str(varval(STRcatalog)));
+/* for compatibility */
+#undef NL_CAT_LOCALE		
 #ifdef NL_CAT_LOCALE /* POSIX-compliant. */
     /*
      * Check if LC_MESSAGES is set in the environment and use it, if so.
