diff --git a/Makefile.in b/Makefile.in
index c6b5f25..ffb93e8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -740,9 +740,20 @@ $(srcdir)/stamp-h.in: $(srcdir)/configure.ac
 
 check test: atconfig $(srcdir)/tests/testsuite
 	$(ENVCMD) - \
-		USER="$(USER)" \
+		USER="$(USER)" PATH="$(PATH)" \
 		$(SHELL) $(srcdir)/tests/testsuite
 
+sys-atconfig: atconfig
+	@gsed -e "s,AUTOTEST_PATH='.'",AUTOTEST_PATH='/usr/gnu/bin:/usr/bin', < \
+	    atconfig > sys-atconfig
+
+system-test: sys-atconfig $(srcdir)/tests/testsuite
+	@mv atconfig atconfig.orig
+	@cp sys-atconfig atconfig
+	$(SHELL) $(srcdir)/tests/testsuite
+	@mv atconfig.orig atconfig
+
+
 #
 # Dependencies
 #
