7010804 Xorg should audit connections and disconnections.

--- xorg-server-21.1.22/configure.ac
+++ xorg-server-21.1.22/configure.ac
@@ -252,6 +252,13 @@ AC_DEFINE(HAVE_EXECINFO_H, 1, [Have exec
     ])]
 )
 
+# Solaris auditing
+AC_CHECK_LIB([bsm], [adt_start_session], [HAVE_LIBBSM=yes ; SYS_LIBS=-lbsm ;
+    AC_DEFINE([HAVE_LIBBSM], [],
+        [Define to 1 if you have the Solaris auditing library (-lbsm)])])
+AM_CONDITIONAL([HAVE_LIBBSM], [test "x$HAVE_LIBBSM" = xyes])
+
+
 dnl ---------------------------------------------------------------------------
 dnl Bus options and CPU capabilities.  Replaces logic in
 dnl hw/xfree86/os-support/bus/Makefile.am, among others.
--- xorg-server-21.1.22/dix/main.c
+++ xorg-server-21.1.22/dix/main.c
@@ -187,6 +187,7 @@ xfont2_init_glyph_caching();
         dixResetRegistry();
         InitFonts();
         InitCallbackManager();
+ 	 OSAuditInit(); /* Must be after InitCallbackManager() */
         InitOutput(&screenInfo, argc, argv);
 
         if (screenInfo.numScreens < 1)
--- xorg-server-21.1.22/include/dix-config.h.in
+++ xorg-server-21.1.22/include/dix-config.h.in
@@ -110,6 +110,9 @@ #undef BUNDLE_ID_PREFIX
 /* Build a standalone xpbproxy */
 #undef STANDALONE_XPBPROXY
 
+/* Define to 1 if you have the Solaris BSM auditing library (-lbsm) */
+#undef HAVE_LIBBSM
+
 /* Define to 1 if you have the `bsd' library (-lbsd). */
 #undef HAVE_LIBBSD
 
--- xorg-server-21.1.22/include/os.h
+++ xorg-server-21.1.22/include/os.h
@@ -713,6 +713,10 @@ LogPrintMarkers(void);
 extern _X_EXPORT void
 xorg_backtrace(void);
 
+/* in osaudit.c */
+extern void
+OSAuditInit (void);
+
 extern _X_EXPORT int
 os_move_fd(int fd);
 
--- xorg-server-21.1.22/os/Makefile.am
+++ xorg-server-21.1.22/os/Makefile.am
@@ -19,6 +19,7 @@ connection.c	\
 	inputthread.c	\
 	io.c		\
 	mitauth.c	\
+	osaudit.c	\
 	oscolor.c	\
 	osdep.h		\
 	osinit.c	\
