7010804 Xorg should audit connections and disconnections.

--- a/configure.ac	2019-02-12 09:47:24.101882768 -0800
+++ b/configure.ac	2019-02-12 09:48:12.625892659 -0800
@@ -249,6 +249,13 @@
     ])]
 )
 
+# Solaris auditing
+AC_CHECK_LIB([bsm], [adt_start_session], [HAVE_LIBBSM=yes ; SYS_LIBS=-lbsm ;
+    AC_DEFINE([HAVE_LIBBSM], [],
+        [Define to 1 if you have the Solaris auditing library (-lbsm)])])
+AM_CONDITIONAL([HAVE_LIBBSM], [test "x$HAVE_LIBBSM" = xyes])
+
+
 dnl ---------------------------------------------------------------------------
 dnl Bus options and CPU capabilities.  Replaces logic in
 dnl hw/xfree86/os-support/bus/Makefile.am, among others.
--- a/dix/main.c	2019-02-12 09:48:27.064674638 -0800
+++ b/dix/main.c	2019-02-12 09:48:44.912497706 -0800
@@ -190,6 +190,7 @@
         dixResetRegistry();
         InitFonts();
         InitCallbackManager();
+ 	 OSAuditInit(); /* Must be after InitCallbackManager() */
         InitOutput(&screenInfo, argc, argv);
 
         if (screenInfo.numScreens < 1)
--- a/include/dix-config.h.in	2019-02-12 09:49:00.713767429 -0800
+++ b/include/dix-config.h.in	2019-02-12 09:49:43.196842484 -0800
@@ -116,6 +116,9 @@
 /* Build a standalone xpbproxy */
 #undef STANDALONE_XPBPROXY
 
+/* Define to 1 if you have the Solaris BSM auditing library (-lbsm) */
+#undef HAVE_LIBBSM
+
 /* Define to 1 if you have the `bsd' library (-lbsd). */
 #undef HAVE_LIBBSD
 
--- a/include/os.h	2019-02-12 09:50:05.297429979 -0800
+++ b/include/os.h	2019-02-12 09:50:48.895959653 -0800
@@ -711,6 +711,10 @@
 extern _X_EXPORT void
 xorg_backtrace(void);
 
+/* in osaudit.c */
+extern void
+OSAuditInit (void);
+
 extern _X_EXPORT int
 os_move_fd(int fd);
 
--- a/os/Makefile.am	2019-02-12 09:51:06.656259950 -0800
+++ b/os/Makefile.am	2019-02-12 09:51:20.344487624 -0800
@@ -19,6 +19,7 @@
 	inputthread.c	\
 	io.c		\
 	mitauth.c	\
+	osaudit.c	\
 	oscolor.c	\
 	osdep.h		\
 	osinit.c	\
