Solaris specific patch

not suitable for upstream

--- xorg-server-21.1.22/os/utils.c
+++ xorg-server-21.1.22/os/utils.c
@@ -903,6 +903,11 @@ }
         else if (strcmp(argv[i], "-reset") == 0) {
             dispatchExceptionAtReset = DE_RESET;
         }
+#ifdef SUNSOFT
+        else if (strcmp(argv[i], "-nobanner") == 0) {
+          /* ignore - just maintaining compatibility with Xsun CLI */
+        }
+#endif
         else if (strcmp(argv[i], "-p") == 0) {
             if (++i < argc)
                 defaultScreenSaverInterval = ((CARD32) atoi(argv[i])) *
