Solaris specific patch

not suitable for upstream

--- a/os/utils.c	2019-02-12 10:30:51.851913598 -0800
+++ b/os/utils.c	2019-02-12 10:32:21.510317440 -0800
@@ -888,6 +888,11 @@
         else if (strcmp(argv[i], "-reset") == 0) {
             dispatchExceptionAtReset = DE_RESET;
         }
+#ifdef SUNSOFT
+        else if (strcmp(argv[i], "-nobanner") == 0) {
+          /* ignore - just maintaining compatibility with Xsun CLI */
+        }
+#endif
         else if (strcmp(argv[i], "-p") == 0) {
             if (++i < argc)
                 defaultScreenSaverInterval = ((CARD32) atoi(argv[i])) *
