Solaris-specific changes for:
Bug 15491721 - SUNBT6721634 The Nvidia logo screen should *not* be displayed

Not suitable for upstream

--- a/hw/xfree86/common/xf86AutoConfig.c        2019-02-12 11:44:32.590848767 -0800
+++ b/hw/xfree86/common/xf86AutoConfig.c        2019-02-12 11:49:57.014895841 -0800
@@ -61,6 +61,18 @@
        "\tIdentifier\t" BUILTIN_DEVICE_NAME "\n" \
        "\tDriver\t\"%s\"\n"

+#ifdef sun
+/*
+ * Allow setting an arbitrary number of options.
+ * Each option should be indented with a tab and newline terminated.
+ */
+#define BUILTIN_DEVICE_SECTION_PRE_OPT \
+       "Section \"Device\"\n" \
+       "\tIdentifier\t" BUILTIN_DEVICE_NAME "\n" \
+       "\tDriver\t\"%s\"\n" \
+       "%s"
+#endif
+
 #define BUILTIN_DEVICE_SECTION_POST \
        "EndSection\n\n"

@@ -68,6 +80,13 @@
        BUILTIN_DEVICE_SECTION_PRE \
        BUILTIN_DEVICE_SECTION_POST

+#ifdef sun
+/* Device section with options */
+#define BUILTIN_DEVICE_SECTION_OPT \
+       BUILTIN_DEVICE_SECTION_PRE_OPT \
+       BUILTIN_DEVICE_SECTION_POST
+#endif
+
 #define BUILTIN_SCREEN_NAME \
        "\"Builtin Default %s Screen %d\""

@@ -184,6 +203,14 @@
     listPossibleVideoDrivers(&md);

     for (i = 0; i < md.nmatches; i++) {
+#ifdef sun
+        const char *nvidia_string = "nvidia";
+        const char *nvidia_opt_no_logo = "\tOption \"NoLogo\" \"True\"\n";
+        if (!strncmp(md.matches[i], nvidia_string, sizeof(nvidia_string)))
+            snprintf(buf, sizeof(buf), BUILTIN_DEVICE_SECTION_OPT,
+                      md.matches[i], 0, md.matches[i], nvidia_opt_no_logo);
+        else
+#endif
         snprintf(buf, sizeof(buf), BUILTIN_DEVICE_SECTION,
                 md.matches[i], 0, md.matches[i]);
         AppendToConfig(buf);

