Solaris specific build patches

not suitable for upstream

--- a/hw/xfree86/Makefile.am	2019-02-12 11:51:41.326033886 -0800
+++ b/hw/xfree86/Makefile.am	2019-02-12 11:52:25.552231067 -0800
@@ -142,7 +142,7 @@
 EXTRA_DIST += sdksyms.sh
 
 sdksyms.dep sdksyms.c: sdksyms.sh
-	$(AM_V_GEN)CPP='$(CPP)' AWK='$(AWK)' $(SHELL) $(srcdir)/sdksyms.sh $(top_srcdir) $(CFLAGS) $(AM_CFLAGS) $(AM_CPPFLAGS)
+	$(AM_V_GEN)CPP='$(CPP)' AWK='$(AWK)' $(SHELL) $(srcdir)/sdksyms.sh $(top_srcdir) $(CFLAGS) $(AM_CFLAGS) $(AM_CPPFLAGS) -I/usr/include -I/usr/include/xorg -I/usr/include/pixman-1 -I/usr/include/X11/dri
 
 SDKSYMS_DEP = sdksyms.dep
 -include $(SDKSYMS_DEP)
--- a/configure.ac	2019-02-12 11:52:44.813937675 -0800
+++ b/configure.ac	2019-02-12 11:53:11.044484824 -0800
@@ -191,6 +191,8 @@
 dnl Find the math libary, then check for cbrt function in it.
 AC_CHECK_LIB(m, sqrt)
 AC_CHECK_FUNCS([cbrt])
+# But we don't want to link everything with libm
+LIBS="${LIBS//-lm}"
 
 dnl AGPGART headers
 AC_CHECK_HEADERS([linux/agpgart.h sys/agpio.h sys/agpgart.h], AGP=yes)
@@ -1713,6 +1715,7 @@
 XSERVER_LIBS="$DIX_LIB $MI_LIB $OS_LIB"
 XSERVER_SYS_LIBS="${XSERVERLIBS_LIBS} ${SYS_LIBS} ${LIBS}"
 XSERVER_SYS_LIBS="${XSERVER_SYS_LIBS} -lproject"
+XSERVER_SYS_LIBS="${XSERVER_SYS_LIBS} -lm"
 AC_SUBST([XSERVER_LIBS])
 AC_SUBST([XSERVER_SYS_LIBS])
 
