
--- a/meson.build	2024-05-04 22:05:20.105625818 -0700
+++ b/meson.build	2024-05-04 22:06:06.512232863 -0700
@@ -21,10 +21,6 @@
 gtk_wayland_dep = dependency('gtk+-wayland-3.0', version: '>= 3.21.5', required: get_option('wayland'))
 gtksourceview_dep = dependency('gtksourceview-4', version: '>=4.0.3')
 harfbuzz_dep = dependency('harfbuzz', version: '>=0.9.9')
-webkit_dep = dependency('webkit2gtk-4.1', required : false)
-if not webkit_dep.found()
-  webkit_dep = dependency('webkit2gtk-4.0')
-endif
 
 bindir = join_paths(get_option('prefix'), get_option('bindir'))
 datadir = join_paths(get_option('prefix'), get_option('datadir'))
