#
# Sudo should prevent itself from killing the world.
#
--- sudo-1.9.17p1/src/exec.c
+++ sudo-1.9.17p1/src/exec.c
@@ -523,7 +523,7 @@ {
     debug_decl(terminate_command, SUDO_DEBUG_EXEC);
 
     /* Avoid killing more than a single process or process group. */
-    if (pid <= 0)
+    if (pid <= 1)
 	debug_return;
 
     /*
--- sudo-1.9.17p1/src/exec_pty.c
+++ sudo-1.9.17p1/src/exec_pty.c
@@ -360,7 +360,8 @@ close(io_fds[SFD_LEADER]);
     }
     sudo_debug_printf(SUDO_DEBUG_NOTICE, "%s: killpg(%d, SIGHUP)",
 	__func__, (int)pgrp);
-    kill(pgrp, SIGHUP);
+    if (pgrp > 1)
+	kill(pgrp, SIGHUP);
 }
 
 /*
