--- a/src/Makefile.am	2018-02-01 11:02:37.082595986 -0800
+++ b/src/Makefile.am	2018-02-01 11:03:39.371445080 -0800
@@ -23,25 +23,4 @@
 #    Eric Anholt <eric@anholt.net>
 
 SUBDIRS = \
-	util \
-	data \
-	demos \
-	egl \
-	fp \
-	fpglsl \
-	glsl \
-        gs \
-	objviewer \
-	osdemos \
-	perf \
-	rbug \
-	redbook \
-	samples \
-	slang \
-	tests \
-	tools \
-	trivial \
-	vp \
-	vpglsl \
-	wgl \
 	xdemos
