From 1fbda192d7ad649f9d2f33684f3c57654f27f72b Mon Sep 17 00:00:00 2001
From: Aleksey Sanin <aleksey@aleksey.com>
Date: Tue, 15 Oct 2019 10:33:50 -0700
Subject: [PATCH] fix tests

---
 tests/testDSig.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/testDSig.sh b/tests/testDSig.sh
index f2014464..dd7db697 100755
--- a/tests/testDSig.sh
+++ b/tests/testDSig.sh
@@ -433,7 +433,7 @@ execDSigTest $res_success \
     "aleksey-xmldsig-01/enveloped-ecdsa-java-bug" \
     "sha512 ecdsa-sha512" \
     "ecdsa x509" \
-    "--trusted-$cert_format $topfolder/keys/enveloped-ecdsa-java-bug-cert.$cert_format --enabled-key-data x509"
+    "--trusted-$cert_format $topfolder/keys/enveloped-ecdsa-java-bug-cert.$cert_format --enabled-key-data x509 --verification-time 2019-01-01+00:00:00"
 fi
 
 #
