See ../../common/patches/042-bn_sqr_mont.patch for details.

--- a/crypto/bn/asm/sparcv9-mont.pl	2021-02-17 16:46:15.059920755 +0000
+++ b/crypto/bn/asm/sparcv9-mont.pl	2021-02-17 16:46:22.368576500 +0000
@@ -115,7 +115,7 @@
 	ld	[$np],$car1		! np[0]
 	sub	%o7,$bias,%sp		! alloca
 	ld	[$np+4],$npj		! np[1]
-	be,pt	SIZE_T_CC,.Lbn_sqr_mont
+	nop				! disable bn_sqr_mont for now
 	mov	12,$j
 
 	mulx	$car0,$mul0,$car0	! ap[0]*bp[0]
