components/lua is 5.4.0 - not suitable for upstream

--- lgi-0.9.2/lgi/Makefile.orig 2019-11-18 04:25:31.096072164 -0800
+++ lgi-0.9.2/lgi/Makefile      2019-11-18 04:25:49.182673007 -0800
@@ -17,17 +17,17 @@
 VERSION_FILE = version.lua

 ifneq ($(filter cygwin% msys% mingw%, $(HOST_OS)),)
-CORE = corelgilua51.dll
+CORE = corelgilua54.dll
 LIBFLAG = -shared
 LIBS += -llua
 else
 ifeq ($(HOST_OS),darwin)
-CORE = corelgilua51.so
+CORE = corelgilua54.so
 LIBFLAG = -bundle -undefined dynamic_lookup
 CCSHARED = -fno-common
 GOBJECT_INTROSPECTION_LIBDIR = $(shell $(PKG_CONFIG) --variable=libdir $(GINAME))
 else
-CORE = corelgilua51.so
+CORE = corelgilua54.so
 LIBFLAG = -shared
 CCSHARED = -fPIC
 endif
--- lgi-0.9.2/lgi/core.c.orig   2019-11-18 04:21:32.095989262 -0800
+++ lgi-0.9.2/lgi/core.c        2019-11-18 04:22:21.583438362 -0800
@@ -684,7 +684,7 @@
 }

 G_MODULE_EXPORT int
-luaopen_lgi_corelgilua51 (lua_State* L)
+luaopen_lgi_corelgilua54 (lua_State* L)
 {
   LgiStateMutex *mutex;
   gint state_id;
--- lgi-0.9.0/lgi/core.lua	2015-03-23 15:48:44.000000000 -0700
+++ lgi-0.9.0/lgi/core.lua	2015-06-25 23:32:50.067448193 -0700
@@ -11,7 +11,7 @@
 -- This module decides what kind of core routines should be loaded.
 -- Currently only one implementation exists, standard-Lua C-side
 -- implementation, LuaJIT-FFI-based one is planned.
-local core = require 'lgi.corelgilua51'
+local core = require 'lgi.corelgilua54'
 
 -- Helper methods for converting between CamelCase and uscore_delim
 -- names.
