https://github.com/php/php-src/issues/9959

--- php-8.1.12/sapi/fpm/fpm/events/port.c
+++ php-8.1.12/sapi/fpm/fpm/events/port.c
@@ -153,6 +153,9 @@
 		/* fire the event */
 		fpm_event_fire((struct fpm_event_s *)events[i].portev_user);
 
+		/* re-associate for next event */
+		fpm_event_port_add(events[i].portev_user);
+
 		/* sanity check */
 		if (fpm_globals.parent_pid != getpid()) {
 			return -2;
