Since SOX is dead upstream, we have eol'd it.
Bug 26046950 - espeak update due to eof of SOX 
Bug 24824502 - sox EOF 

--- espeak-ng-1.51.1/src/libespeak-ng/soundicon.c
+++ espeak-ng-1.51.1/src/libespeak-ng/soundicon.c
@@ -71,6 +71,7 @@ }
 	fname_temp[0] = 0;
 
 	f = NULL;
+#if !defined(__sun) // no sox
 	if ((f = fopen(fname, "rb")) != NULL) {
 		int ix;
 		int fd_temp;
@@ -104,6 +105,7 @@ if (system(command) == 0)
 				fname = fname_temp;
 		}
 	}
+#endif
 
 	if (f == NULL) {
 		f = fopen(fname, "rb");
