--- espeak-ng-1.50-0577cd9/src/libespeak-ng/mbrowrap.c.~1~	Mon Jan  3 11:14:18 2022
+++ espeak-ng-1.50-0577cd9/src/libespeak-ng/mbrowrap.c	Thu May 12 05:27:52 2022
@@ -216,7 +216,11 @@
 		_exit(1);
 	}
 
+#if defined(__sun) && defined(__SVR4)
+	snprintf(charbuf, sizeof(charbuf), "/proc/%d/psinfo", mbr_pid);
+#else
 	snprintf(charbuf, sizeof(charbuf), "/proc/%d/stat", mbr_pid);
+#endif
 	mbr_proc_stat = open(charbuf, O_RDONLY);
 	if (mbr_proc_stat == -1) {
 		error = errno;
@@ -410,8 +414,21 @@
 	return result;
 }
 
+#if defined(__sun) && defined(__SVR4) /* Solaris */
+#include <procfs.h>
 static int mbrola_is_idle(void)
 {
+	psinfo_t ps;
+
+	// look in /proc to determine if mbrola is still running or sleeping
+	if (pread(mbr_proc_stat, &ps, sizeof(ps), 0) != sizeof(ps))
+		return 0;
+
+	return strcmp(ps.pr_fname, "mbrola") == 0 && ps.pr_lwp.pr_sname == 'S';
+}
+#else
+static int mbrola_is_idle(void)
+{
 	char *p;
 	char buffer[20]; // looking for "12345 (mbrola) S" so 20 is plenty
 
@@ -425,6 +442,7 @@
 		return 0;
 	return p[1] == ' ' && p[2] == 'S';
 }
+#endif
 
 static ssize_t receive_from_mbrola(void *buffer, size_t bufsize)
 {
