Allows to use Googletest from sources instead from system.

--- c-ares-1.34.5/test/CMakeLists.txt
+++ c-ares-1.34.5/test/CMakeLists.txt
@@ -7,7 +7,7 @@ SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
 ENDIF ()
 
 find_package(Threads)
-find_package(GTest REQUIRED)
+add_subdirectory(googletest)
 
 IF (CMAKE_VERSION VERSION_LESS "3.23.0")
   Message("GoogleTest found, cmake version too old to verify GMock, look for errors...")
