Solaris pkgconfig file changes

Not suitable for upstream

--- a/glib/meson.build	2019-10-31 16:01:22.777037273 -0700
+++ b/glib/meson.build	2019-10-31 16:01:38.923014840 -0700
@@ -463,12 +463,9 @@
   subdirs : ['glib-2.0'],
   extra_cflags : ['-I${libdir}/glib-2.0/include'] + win32_cflags,
   variables : [
-    'glib_genmarshal=' + '${bindir}' / 'glib-genmarshal',
-    'gobject_query=' + '${bindir}' / 'gobject-query',
-    'glib_mkenums=' + '${bindir}' / 'glib-mkenums',
-    'glib_valgrind_suppressions=' + '${datadir}' /
-      valgrind_suppression_file_install_subdir /
-      fs.name(valgrind_suppression_file),
+    'glib_genmarshal=glib-genmarshal',
+    'gobject_query=gobject-query',
+    'glib_mkenums=glib-mkenums',
   ],
   uninstalled_variables : [
     'glib_genmarshal=${prefix}/gobject/glib-genmarshal',
--- a/gio/meson.build   2025-07-17 17:30:29.345634475 -0700
+++ b/gio/meson.build   2025-07-17 17:34:48.172102102 -0700
@@ -904,14 +904,14 @@
     'schemasdir=' + '${datadir}' / schemas_subdir,
     'dtdsdir=' + '${datadir}' / dtds_subdir,
     'giomoduledir=' + pkgconfig_giomodulesdir,
-    'gio=' + '${bindir}' / 'gio',
-    'gio_querymodules=' + pkgconfig_multiarch_bindir / 'gio-querymodules',
-    'glib_compile_schemas=' + pkgconfig_multiarch_bindir / 'glib-compile-schemas',
-    'glib_compile_resources=' + '${bindir}' / 'glib-compile-resources',
-    'gdbus=' + '${bindir}' /'gdbus',
-    'gdbus_codegen=' + '${bindir}' / 'gdbus-codegen',
-    'gresource=' + '${bindir}' / 'gresource',
-    'gsettings=' + '${bindir}' / 'gsettings',
+    'gio=gio',
+    'gio_querymodules=gio-querymodules',
+    'glib_compile_schemas=glib-compile-schemas',
+    'glib_compile_resources=glib-compile-resources',
+    'gdbus=gdbus',
+    'gdbus_codegen=gdbus-codegen',
+    'gresource=gresource',
+    'gsettings=gsettings',   
   ],
   uninstalled_variables : [
     'gio=${prefix}/gio/gio',
--- a/meson.build       2025-07-17 17:35:11.758075818 -0700
+++ b/meson.build       2025-07-17 17:35:33.872339073 -0700
@@ -2125,7 +2125,7 @@
   glibconfig_conf.set('g_threads_impl_def', 'WIN32')
   glib_conf.set('THREADS_WIN32', 1)
 else
-  thread_dep = dependency('threads')
+  thread_dep = []
   threads_implementation = 'posix'
   pthread_prefix = '''
       #ifndef _GNU_SOURCE
