diff --git a/nss/coreconf/config.mk b/nss/coreconf/config.mk
index 60a0841..481cc56 100644
--- a/nss/coreconf/config.mk
+++ b/nss/coreconf/config.mk
@@ -207,3 +207,5 @@ DEFINES += -DNO_NSPR_10_SUPPORT
 
 # Hide old, deprecated, TLS cipher suite names when building NSS
 DEFINES += -DSSL_DISABLE_DEPRECATED_CIPHER_SUITE_NAMES
+
+INCLUDES := $(subst -I/usr/include/mps,,$(INCLUDES)) -I/usr/include/mps
