# Filed upstream as https://github.com/Perl/perl5/issues/22138

--- perl-5.38.2/cpan/Socket/Socket.xs
+++ perl-5.38.2/cpan/Socket/Socket.xs
@@ -938,7 +938,7 @@ unpack_sockaddr_un(sun_sv)
 	if (!SvOK(sun_sv))
 	    croak("Undefined address for %s", "Socket::unpack_sockaddr_un");
 	sun_ad = SvPVbyte(sun_sv,sockaddrlen);
-#   if defined(__linux__) || defined(__CYGWIN__) || defined(HAS_SOCKADDR_SA_LEN)
+#   if defined(__linux__) || defined(__CYGWIN__) || defined(__sun__) || defined(HAS_SOCKADDR_SA_LEN)
 	/* On Linux, Cygwin or *BSD sockaddrlen on sockets returned by accept,
 	 * recvfrom, getpeername and getsockname is not equal to sizeof(addr). */
 	if (sockaddrlen < sizeof(addr)) {
