We have gweather 4 and geocode-glib-2.0 (built with libsoup3)

--- a/gnome-initial-setup/meson.build	2024-01-23 13:40:15.376435528 -0800
+++ b/gnome-initial-setup/meson.build	2024-01-23 13:42:04.182274400 -0800
@@ -33,7 +33,7 @@
     dependency ('gnome-desktop-3.0', version: '>= 3.7.5'),
     dependency ('gsettings-desktop-schemas', version: '>= 3.37.1'),
     dependency ('fontconfig'),
-    dependency ('gweather-3.0'),
+    dependency ('gweather4'),
     dependency ('goa-1.0'),
     dependency ('goa-backend-1.0'),
     dependency ('gtk+-3.0', version: '>= 3.11.3'),
@@ -40,7 +40,7 @@
     dependency ('glib-2.0', version: '>= 2.63.1'),
     dependency ('gio-unix-2.0', version: '>= 2.53.0'),
     dependency ('gdm', version: '>= 3.8.3'),
-    dependency ('geocode-glib-1.0'),
+    dependency ('geocode-glib-2.0'),
     dependency ('libgeoclue-2.0', version: '>= 2.3.1'),
     cc.find_library('m', required: false),
     dependency ('pango', version: '>= 1.32.5'),
