disable gnome-online-accounts registration page

--- a/gnome-initial-setup/pages/meson.build	2024-05-04 04:08:04.735154365 +0000
+++ b/gnome-initial-setup/pages/meson.build	2024-05-04 04:08:23.915394954 +0000
@@ -2,7 +2,6 @@
    'language',
    'keyboard',
    'privacy',
-   'goa',
    'software',
    'summary',
    'welcome',
--- a/gnome-initial-setup/gnome-initial-setup.c	2024-05-04 04:13:45.223006786 +0000
+++ a/gnome-initial-setup/gnome-initial-setup.c	2024-05-04 04:13:58.190800890 +0000
@@ -70,7 +70,6 @@
   PAGE (keyboard, FALSE),
   PAGE (privacy,  FALSE),
   PAGE (software, TRUE),
-  PAGE (goa,      FALSE),
 #ifdef HAVE_PARENTAL_CONTROLS
   PAGE (parental_controls, TRUE),
   PAGE (parent_password, TRUE),
--- a/gnome-initial-setup/meson.build	2024-05-04 22:02:23.649090257 -0700
+++ b/gnome-initial-setup/meson.build	2024-05-04 22:02:30.070505149 -0700
@@ -35,7 +35,6 @@
     dependency ('fontconfig'),
     dependency ('gweather4'),
     dependency ('goa-1.0'),
-    dependency ('goa-backend-1.0'),
     dependency ('gtk+-3.0', version: '>= 3.11.3'),
     dependency ('glib-2.0', version: '>= 2.63.1'),
     dependency ('gio-unix-2.0', version: '>= 2.53.0'),
