disable gnome-online-accounts registration page

--- gnome-initial-setup-41.4/gnome-initial-setup/pages/meson.build
+++ gnome-initial-setup-41.4/gnome-initial-setup/pages/meson.build
@@ -2,7 +2,6 @@ pages = [
    'language',
    'keyboard',
    'privacy',
-   'goa',
    'software',
    'summary',
    'welcome',
--- gnome-initial-setup-41.4/gnome-initial-setup/gnome-initial-setup.c
+++ gnome-initial-setup-41.4/gnome-initial-setup/gnome-initial-setup.c
@@ -70,7 +70,6 @@ PAGE (language, FALSE),
   PAGE (keyboard, FALSE),
   PAGE (privacy,  FALSE),
   PAGE (software, TRUE),
-  PAGE (goa,      FALSE),
 #ifdef HAVE_PARENTAL_CONTROLS
   PAGE (parental_controls, TRUE),
   PAGE (parent_password, TRUE),
--- gnome-initial-setup-41.4/gnome-initial-setup/meson.build
+++ gnome-initial-setup-41.4/gnome-initial-setup/meson.build
@@ -33,8 +33,6 @@ dependency ('accountsservice'),
     dependency ('gnome-desktop-3.0', version: '>= 3.7.5'),
     dependency ('gsettings-desktop-schemas', version: '>= 3.37.1'),
     dependency ('fontconfig'),
-    dependency ('goa-1.0'),
-    dependency ('goa-backend-1.0'),
     dependency ('gtk+-3.0', version: '>= 3.11.3'),
     dependency ('glib-2.0', version: '>= 2.63.1'),
     dependency ('gio-unix-2.0', version: '>= 2.53.0'),
