requires pipewire to work

--- a/js/dbusServices/meson.build	2024-02-04 08:54:41.615486149 -0800
+++ b/js/dbusServices/meson.build	2024-02-04 08:55:05.047894280 -0800
@@ -6,7 +6,6 @@
 dbus_services = {
   'org.gnome.Shell.Extensions': 'extensions',
   'org.gnome.Shell.Notifications': 'notifications',
-  'org.gnome.Shell.Screencast': 'screencast',
   'org.gnome.ScreenSaver': 'screensaver',
 }
 
--- a/data/dbus-interfaces/meson.build	2024-02-04 09:00:30.533146330 -0800
+++ b/data/dbus-interfaces/meson.build	2024-02-04 09:00:38.155368390 -0800
@@ -2,7 +2,6 @@
   'org.gnome.Shell.Extensions.xml',
   'org.gnome.Shell.Introspect.xml',
   'org.gnome.Shell.PadOsd.xml',
-  'org.gnome.Shell.Screencast.xml',
   'org.gnome.Shell.Screenshot.xml',
   'org.gnome.ShellSearchProvider.xml',
   'org.gnome.ShellSearchProvider2.xml'
