We need to have the same so version 

--- a/src/Makefile.am	2018-01-11 19:17:40.094678442 -0800
+++ b/src/Makefile.am	2018-01-11 19:18:18.435810597 -0800
@@ -20,7 +20,7 @@
 AM_CFLAGS = $(CWARNFLAGS) @XKBFILE_CFLAGS@
 
 libxkbfile_la_LIBADD = @XKBFILE_LIBS@
-libxkbfile_la_LDFLAGS = -version-number 1:0:2 -no-undefined
+libxkbfile_la_LDFLAGS = -version-number 5 -no-undefined
 
 libxkbfileincludedir = $(includedir)/X11/extensions
 libxkbfileinclude_HEADERS = \
