fix build issues with vala

--- a/lib/math-equation.vala	2021-08-06 19:23:13.199178715 -0700
+++ b/lib/math-equation.vala	2021-08-06 19:26:08.465733250 -0700
@@ -826,7 +826,7 @@
             get_iter_at_mark (out iter_prev, get_insert ());
             if (iter_prev.backward_char () &&  iter_prev.get_char ().to_string () == "×" )
             {
-                (this as Gtk.TextBuffer).backspace (iter, true, true);
+                (this as Gtk.TextBuffer).backspace (ref iter, true, true);
                 insert_at_cursor ("^", -1);
                 return;
             }
@@ -840,7 +840,7 @@
             get_iter_at_mark (out iter_prev, get_insert ());
             if (iter_prev.backward_char () &&  iter_prev.get_char ().to_string () == ">" )
             {
-                (this as Gtk.TextBuffer).backspace (iter, true, true);
+                (this as Gtk.TextBuffer).backspace (ref iter, true, true);
                 insert_at_cursor ("»", -1);
                 return;
             }
@@ -854,7 +854,7 @@
             get_iter_at_mark (out iter_prev, get_insert ());
             if (iter_prev.backward_char () &&  iter_prev.get_char ().to_string () == "<" )
             {
-                (this as Gtk.TextBuffer).backspace (iter, true, true);
+                (this as Gtk.TextBuffer).backspace (ref iter, true, true);
                 insert_at_cursor ("«", -1);
                 return;
             }
@@ -1303,7 +1303,7 @@
         {
             Gtk.TextIter iter;
             get_iter_at_mark (out iter, get_insert ());
-            (this as Gtk.TextBuffer).backspace (iter, true, true);
+            (this as Gtk.TextBuffer).backspace (ref iter, true, true);
         }
     }
 
