GCC 14 fixes.

--- infiniband-diags-1.6.5/src/ibping.c
+++ infiniband-diags-1.6.5/src/ibping.c
@@ -68,7 +68,7 @@
 	s += n + 1;
 	sz -= n + 1;
 
-	if (getdomainname(s, sz) < 0)
+	// Right getdomainname() is not avaiable on Solaris // if (getdomainname(s, sz) < 0)
 		snprintf(s, sz, "?domainname?");
 	if (strlen(s) == 0)
 		s[-1] = 0;	/* no domain */
--- infiniband-diags-1.6.5/src/ibsysstat.c
+++ infiniband-diags-1.6.5/src/ibsysstat.c
@@ -135,7 +135,7 @@
 		s += n + 1;
 		sz -= n + 1;
 		ret += n + 1;
-		if (getdomainname(s, sz) < 0)
+		// Right getdomainname() is not avaiable on Solaris // if (getdomainname(s, sz) < 0)
 			snprintf(s, sz, "?domainname?");
 		if ((n = strlen(s)) == 0)
 			s[-1] = 0;	/* no domain */
--- infiniband-diags-1.6.5/src/ibsysstat.c
+++ infiniband-diags-1.6.5/src/ibsysstat.c
@@ -39,6 +39,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <strings.h>  // for bzero()
 #include <getopt.h>
 
 #include <infiniband/umad.h>
