This test is missing from the tarball!

Inform upstream.

diff -r b272467518e7 libsm/Makefile.m4
--- a/libsm/Makefile.m4	Wed Mar 08 09:19:45 2023 -0800
+++ b/libsm/Makefile.m4	Wed Mar 08 09:37:16 2023 -0800
@@ -45,7 +45,6 @@
 smcheck(`t-ixlen', `compile')
 smcheck(`t-ixlen.sh', `run')
 smcheck(`t-streq', `compile')
-smcheck(`t-utf8_valid', `compile')
 smcheck(`t-streq.sh', `run')
 divert(bldTARGETS_SECTION)
 divert(0)
