--- ntpd/ntpd.c
+++ ntpd/ntpd.c
@@ -898,17 +902,10 @@ ntpdmain(
 		msyslog(LOG_NOTICE, "%s", buf);
 	}
 
-	msyslog(LOG_NOTICE, "----------------------------------------------------");
-	msyslog(LOG_NOTICE, "ntp-4 is maintained by Network Time Foundation,");
-	msyslog(LOG_NOTICE, "Inc. (NTF), a non-profit 501(c)(3) public-benefit");
-	msyslog(LOG_NOTICE, "corporation.  Support and training for ntp-4 are");
-	msyslog(LOG_NOTICE, "available at https://www.nwtime.org/support");
-	msyslog(LOG_NOTICE, "----------------------------------------------------");
-#ifdef DEBUG
-	msyslog(LOG_NOTICE, "DEBUG behavior is enabled - a violation of any"
-			    " diagnostic assertion will cause %s to abort", 
-			    progname);
-#endif
+	msyslog(LOG_NOTICE, "ntp-4 is maintained by Network Time Foundation,"
+	    " Inc. (NTF), a non-profit 501(c)(3) public-benefit"
+	    " corporation.  Support and training for ntp-4 are"
+	    " available at https://www.nwtime.org/support");
 
 	ssl_check_version();
 
