The optional railroad module is not available on Solaris.

--- pyparsing-3.0.9/tests/test_unit.py
+++ pyparsing-3.0.9/tests/test_unit.py
@@ -9291,6 +9291,7 @@ class Test02_WithoutPackrat(ppt.TestPars
         print(expr)
         self.assertEqual("(0-9)", repr(expr))
 
+    @unittest.skip("railroad is not available on Solaris")
     def testEmptyExpressionsAreHandledProperly(self):
         from pyparsing.diagram import to_railroad
 
