The optional railroad module is not available on Solaris.

--- pyparsing-3.2.0/tests/test_unit.py
+++ pyparsing-3.2.0/tests/test_unit.py
@@ -10506,6 +10506,7 @@ class Test02_WithoutPackrat(ppt.TestPars
         print(expr)
         self.assertEqual("(0-9)", repr(expr))
 
+    @unittest.skip("railroad is not available on Solaris")
     def testEmptyExpressionsAreHandledProperly(self):
         from pyparsing.diagram import to_railroad
 
--- pyparsing-3.2.0/tests/test_examples.py
+++ pyparsing-3.2.0/tests/test_examples.py
@@ -43,6 +43,7 @@ class TestExamples(unittest.TestCase):
     def test_booleansearchparser(self):
         self._run("booleansearchparser")
 
+    @unittest.skip("railroad is not available on Solaris")
     def test_rosettacode(self):
         self._run("rosettacode")
 
@@ -52,6 +53,7 @@ class TestExamples(unittest.TestCase):
     def test_lucene_grammar(self):
         self._run("lucene_grammar")
 
+    @unittest.skip("railroad is not available on Solaris")
     def test_range_check(self):
         self._run("range_check")
 
