Changes for Solaris builds.  Not suitable for upstream

--- a/cmake/StaticBuild.cmake	2020-01-31 23:17:43.467000788 -0800
+++ b/cmake/StaticBuild.cmake	2020-01-31 23:18:00.924420953 -0800
@@ -7,6 +7,11 @@
 # a lot of details change with each different build environment.
 #
 
+SET(CMAKE_C_ARCHIVE_CREATE "<CMAKE_AR> <LINK_FLAGS> crS <TARGET> <OBJECTS>")
+SET(CMAKE_C_ARCHIVE_APPEND "<CMAKE_AR> <LINK_FLAGS> r <TARGET> <OBJECTS>")
+SET(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> <LINK_FLAGS> crS <TARGET> <OBJECTS>")
+SET(CMAKE_CXX_ARCHIVE_APPEND "<CMAKE_AR> <LINK_FLAGS> r <TARGET> <OBJECTS>")
+
 option(BUILD_STATIC
     "Link statically against most libraries, if possible" OFF)
 
--- a/unix/tx/TXDialog.h        2025-05-20 15:27:54.618465335 -0700
+++ b/unix/tx/TXDialog.h        2025-05-20 15:28:04.387189143 -0700
@@ -32,6 +32,7 @@
 
 #include "TXWindow.h"
 #include <errno.h>
+#include <cstring>

 class TXDialog : public TXWindow, public TXDeleteWindowCallback {
 public:
--- a/unix/xserver/hw/vnc/Makefile.am	2025-05-20 18:22:01.948328965 -0700
+++ b/unix/xserver/hw/vnc/Makefile.am	2025-05-20 18:26:38.598282570 -0700
@@ -32,7 +32,10 @@
 	vncInput.c vncInputXKB.c qnum_to_xorgevdev.c qnum_to_xorgkbd.c
 
 libvnccommon_la_CPPFLAGS = $(AM_CPPFLAGS) \
-	-I$(TIGERVNC_SRCDIR)/unix/vncconfig
+	-I$(TIGERVNC_SRCDIR)/unix/vncconfig \
+	$(XVNC_CPPFLAGS) ${XSERVERLIBS_CFLAGS} \
+	-I$(includedir) -I/usr/include/xorg \
+	-I$(top_srcdir)/include
 
 bin_PROGRAMS = Xvnc
 
@@ -60,9 +63,11 @@
 	$(LOCAL_LIBS) \
 	$(XSERVER_SYS_LIBS) \
 	$(XVNC_SYS_LIBS) \
-	$(LIBXCVT_LIBS)
+	$(LIBXCVT_LIBS) \
+	-ljpeg -lz -lX11 ../../IA/libIA.la
 
 Xvnc_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+Xvnc_LDFLAGS += $(GNUTLS_LDFLAGS)
 
 if DRI3
 Xvnc_SOURCES += vncDRI3.h vncDRI3.c vncDRI3Draw.c
@@ -72,6 +77,7 @@
 Xvnc_LDADD += $(GBM_LIBS)
 endif
 
+if XORG
 libvnc_la_LTLIBRARIES = libvnc.la
 libvnc_ladir = $(moduledir)/extensions
 
@@ -81,12 +87,17 @@
 nodist_EXTRA_libvnc_la_SOURCES = dummy.cxx
 
 libvnc_la_CPPFLAGS = $(AM_CPPFLAGS) \
+	$(XVNC_CPPFLAGS) \
 	-I$(top_srcdir)/hw/xfree86/common \
 	-I$(top_srcdir)/hw/xfree86/os-support \
-	-I$(top_srcdir)/hw/xfree86/os-support/bus
+	-I$(top_srcdir)/hw/xfree86/os-support/bus \
+	-I$(top_srcdir)/include \
+	${XSERVERLIBS_CFLAGS} \
+	-I/usr/include/xorg
 
 libvnc_la_LDFLAGS = -module -avoid-version -Wl,-z,now
 
 libvnc_la_LIBADD = libvnccommon.la $(COMMON_LIBS)
+endif
 
 EXTRA_DIST = Xvnc.man
