Changes for Solaris builds.  Not suitable for upstream

--- a/unix/xserver/hw/vnc/Makefile.am	2020-01-31 23:13:37.135370809 -0800
+++ b/unix/xserver/hw/vnc/Makefile.am	2020-01-31 23:16:58.882282063 -0800
@@ -23,7 +23,7 @@
 
 libvnccommon_la_CPPFLAGS = -DVENDOR_RELEASE="$(VENDOR_RELEASE)" -I$(TIGERVNC_SRCDIR)/unix/common \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common -UHAVE_CONFIG_H \
-	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) ${XSERVERLIBS_CFLAGS} \
+	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) ${XSERVERLIBS_CFLAGS} -I$(includedir) -I/usr/include/xorg \
 	-I$(top_srcdir)/include
 
 bin_PROGRAMS = Xvnc
@@ -64,10 +68,12 @@
 	-I$(top_srcdir)/hw/xfree86/os-support \
 	-I$(top_srcdir)/hw/xfree86/os-support/bus \
 	-I$(top_srcdir)/include \
-	${XSERVERLIBS_CFLAGS}
+	${XSERVERLIBS_CFLAGS} \
+	-I/usr/include/xorg
 
 libvnc_la_LDFLAGS = -module -avoid-version -Wl,-z,now
 
 libvnc_la_LIBADD = libvnccommon.la $(COMMON_LIBS)
+endif
 
 EXTRA_DIST = Xvnc.man
--- a/cmake/StaticBuild.cmake	2020-01-31 23:17:43.467000788 -0800
+++ b/cmake/StaticBuild.cmake	2020-01-31 23:18:00.924420953 -0800
@@ -7,6 +7,11 @@
 # a lot of details change with each different build environment.
 #
 
+SET(CMAKE_C_ARCHIVE_CREATE "<CMAKE_AR> <LINK_FLAGS> crS <TARGET> <OBJECTS>")
+SET(CMAKE_C_ARCHIVE_APPEND "<CMAKE_AR> <LINK_FLAGS> r <TARGET> <OBJECTS>")
+SET(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> <LINK_FLAGS> crS <TARGET> <OBJECTS>")
+SET(CMAKE_CXX_ARCHIVE_APPEND "<CMAKE_AR> <LINK_FLAGS> r <TARGET> <OBJECTS>")
+
 option(BUILD_STATIC
     "Link statically against most libraries, if possible" OFF)
 
--- a/unix/xserver/hw/vnc/Makefile.am	2023-03-28 10:30:20.681002781 -0700
+++ b/unix/xserver/hw/vnc/Makefile.am	2023-03-28 10:32:32.502191397 -0700
@@ -42,7 +42,8 @@
 	-UHAVE_CONFIG_H \
 	-DXFree86Server -DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common -I$(TIGERVNC_SRCDIR)/unix/common \
-	-I$(top_srcdir)/include ${XSERVERLIBS_CFLAGS}
+	-I$(top_srcdir)/include ${XSERVERLIBS_CFLAGS} \
+	-I/usr/include/xorg
 
 LOCAL_LIBS = \
 	$(XVNC_LIBS) \
@@ -51,10 +52,13 @@
 	$(COMMON_LIBS)
 
 Xvnc_DEPENDENCIES = $(LOCAL_LIBS)
-Xvnc_LDADD = $(LOCAL_LIBS) $(XSERVER_SYS_LIBS) $(XVNC_SYS_LIBS) -lX11
+Xvnc_LDADD = $(LOCAL_LIBS) $(XSERVER_SYS_LIBS) $(XVNC_SYS_LIBS) -lX11 \
+	     -ljpeg -lz ../../IA/libIA.la 
 
 Xvnc_LDFLAGS = $(LD_EXPORT_SYMBOLS_FLAG)
+Xvnc_LDFLAGS += $(GNUTLS_LDFLAGS)
 
+if XORG
 libvnc_la_LTLIBRARIES = libvnc.la
 libvnc_ladir = $(moduledir)/extensions
 
