Patch for upstream: fixes core dump issue with columns == 0.

diff -r 08f5a6355d28 -r b750a1b08f07 src/paps.c
--- a/src/paps.c	Thu Feb 18 09:25:09 2016 +0900
+++ b/src/paps.c	Thu Feb 18 09:25:10 2016 +0900
@@ -679,6 +679,11 @@
     num_columns = 1;
   }
 
+  if (num_columns == 0) {
+    fprintf(stderr, _("%s: Invalid input: --columns=%d, using default.\n"), g_get_prgname (), num_columns);
+    num_columns = 1;
+  } 
+
   if (num_columns == 1)
     total_gutter_width = 0;
   else
