Add libraries needed for build to succeed with -z defs

--- xf86-video-mga-2.0.0/src/Makefile.am
+++ xf86-video-mga-2.0.0/src/Makefile.am
@@ -27,7 +27,9 @@ AM_CFLAGS = @XORG_CFLAGS@ @DRI_CFLAGS@
 mga_drv_la_LTLIBRARIES = mga_drv.la
 mga_drv_la_LDFLAGS = -module -avoid-version
 mga_drv_ladir = @moduledir@/drivers
-mga_drv_la_LIBADD = $(noinst_LTLIBRARIES)
+mga_drv_la_LIBADD = $(noinst_LTLIBRARIES) $(XORG_LIBS) $(PCIACCESS_LIBS)
+mga_drv_la_LIBADD += -L$(moduledir) -lfb -lpixman-1 -lshadowfb -lfbdevhw -lexa
+mga_drv_la_LIBADD += -lvgahw -lvbe -lint10
 
 mga_drv_la_SOURCES = \
          mga_bios.c \
