For Solaris data type uint64_t has to be defined.

--- a/src/cups/backend_common.h
+++ b/src/cups/backend_common.h
@@ -49,6 +49,10 @@ #ifdef ERROR
 #undef ERROR
 #endif
 
+#if defined (__SOLARIS__) || defined(__sun) 
+#define __uint64_t  uint64_t
+#endif 
+
 #define STR_LEN_MAX 64
 #define STATE( ... ) do { if (!quiet) fprintf(logger, "STATE: " __VA_ARGS__ ); } while(0)
 #define ATTR( ... ) do { if (!quiet) fprintf(logger, "ATTR: " __VA_ARGS__ ); } while(0)
