--- old/src/common/debug.c	2023-12-12 12:55:40.962057265 +0100
+++ new/src/common/debug.c	2023-12-12 12:57:23.122355187 +0100
@@ -136,7 +136,7 @@ oscap_verbosity_levels oscap_verbosity_l
 bool oscap_set_verbose(const char *verbosity_level, const char *filename)
 {
 	if (verbosity_level == NULL) {
-		verbosity_level = "WARNING";
+		return true;
 	}
 	__debuglog_level = oscap_verbosity_level_from_cstr(verbosity_level);
 	if (__debuglog_level == DBG_UNKNOWN) {
@@ -212,7 +212,11 @@ static void debug_message_start(int leve
 	default:
 		l = '0';
 	}
+#if !defined(__SVR4) && !defined (__sun)
 	fprintf(__debuglog_fp, "%c: %s: ", l, GET_PROGRAM_NAME);
+#else
+	fprintf(__debuglog_fp, "%c: %ld: ", l, (long) getpid());
+#endif
 	for (int i = 0; i < indent; i++) {
 		fprintf(__debuglog_fp, "  ");
 	}
@@ -235,12 +239,17 @@ static void debug_message_devel_metadata
 	/* XXX: non-portable usage of pthread_t */
 	unsigned long long tid = (unsigned long long) thread;
 #endif
+#if !defined(__SVR4) && !defined (__sun)
 	fprintf(__debuglog_fp, " [%s(%ld):%s(%llx):%s:%zu:%s]",
 		GET_PROGRAM_NAME, (long) getpid(), thread_name,
 		tid, f, line, fn);
 #else
 	fprintf(__debuglog_fp, " [%ld:%s:%zu:%s]", (long) getpid(),
 		f, line, fn);
+#endif
+#else
+	fprintf(__debuglog_fp, " [%ld:%s:%zu:%s]", (long) getpid(),
+		f, line, fn);
 #endif
 }
 
