diff -ruN openscap-1.3.4.org/CMakeLists.txt openscap-1.3.4/CMakeLists.txt
--- openscap-1.3.4.org/CMakeLists.txt	2021-03-01 22:46:29.603108127 +0000
+++ openscap-1.3.4/CMakeLists.txt	2021-03-01 22:47:18.168252627 +0000
@@ -291,6 +291,7 @@
 # SOLARIS PROBES
 cmake_dependent_option(OPENSCAP_PROBE_SOLARIS_ISAINFO "Solaris isainfo probe" ON "ENABLE_PROBES_SOLARIS" OFF)
 cmake_dependent_option(OPENSCAP_PROBE_SOLARIS_SMF "Solaris smf probe" ON "ENABLE_PROBES_SOLARIS" OFF)
+cmake_dependent_option(OPENSCAP_PROBE_SOLARIS_SMFPROPERTY "Solaris smf probe" ON "ENABLE_PROBES_SOLARIS" OFF)
 
 # WINDOWS PROBES
 cmake_dependent_option(OPENSCAP_PROBE_WINDOWS_ACCESSTOKEN "Windows accesstoken probe" ON "ENABLE_PROBES_WINDOWS" OFF)
@@ -421,6 +422,7 @@
 message(STATUS "Solaris probes: ${ENABLE_PROBES_SOLARIS}")
 message(STATUS "  Solaris isainfo probe: ${OPENSCAP_PROBE_SOLARIS_ISAINFO}")
 message(STATUS "  Solaris smf probe: ${OPENSCAP_PROBE_SOLARIS_SMF}")
+message(STATUS "  Solaris smfproperty probe: ${OPENSCAP_PROBE_SOLARIS_SMFPROPERTY}")
 message(STATUS " ")
 
 
diff -ruN openscap-1.3.4.org/config.h.in openscap-1.3.4/config.h.in
--- openscap-1.3.4.org/config.h.in	2021-03-01 22:46:29.514608087 +0000
+++ openscap-1.3.4/config.h.in	2021-03-01 22:47:18.168699368 +0000
@@ -111,6 +111,7 @@
 #cmakedefine OPENSCAP_PROBE_LINUX_SYSTEMDUNITPROPERTY
 #cmakedefine OPENSCAP_PROBE_SOLARIS_ISAINFO
 #cmakedefine OPENSCAP_PROBE_SOLARIS_SMF
+#cmakedefine OPENSCAP_PROBE_SOLARIS_SMFPROPERTY
 #cmakedefine OPENSCAP_PROBE_UNIX_DNSCACHE
 #cmakedefine OPENSCAP_PROBE_UNIX_FILE
 #cmakedefine OPENSCAP_PROBE_UNIX_FILEEXTENDEDATTRIBUTE
diff -ruN openscap-1.3.4.org/src/OVAL/oval_enumerations.c openscap-1.3.4/src/OVAL/oval_enumerations.c
--- openscap-1.3.4.org/src/OVAL/oval_enumerations.c	2021-03-01 22:46:29.535840530 +0000
+++ openscap-1.3.4/src/OVAL/oval_enumerations.c	2021-03-01 22:47:18.169258373 +0000
@@ -547,6 +547,7 @@
 	{OVAL_SOLARIS_PATCH, "patch"},
 	{OVAL_SOLARIS_PATCH54, "patch54"},
 	{OVAL_SOLARIS_SMF, "smf"},
+	{OVAL_SOLARIS_SMFPROPERTY, "smfproperty"},
 	{OVAL_SOLARIS_NDD, "ndd"},
 	{OVAL_SOLARIS_PACKAGECHECK, "packagecheck"},
 	{OVAL_SUBTYPE_UNKNOWN, NULL}
diff -ruN openscap-1.3.4.org/src/OVAL/probes/probe-table.c openscap-1.3.4/src/OVAL/probes/probe-table.c
--- openscap-1.3.4.org/src/OVAL/probes/probe-table.c	2021-03-01 22:46:29.552014133 +0000
+++ openscap-1.3.4/src/OVAL/probes/probe-table.c	2021-03-01 22:47:18.169745230 +0000
@@ -135,6 +135,10 @@
 #include "unix/solaris/smf.h"
 #endif
 
+#ifdef OPENSCAP_PROBE_SOLARIS_SMFPROPERTY
+#include "unix/solaris/smfproperty.h"
+#endif
+
 #ifdef OPENSCAP_PROBE_UNIX_DNSCACHE
 #include "unix/dnscache_probe.h"
 #endif
@@ -298,6 +302,9 @@
 #ifdef OPENSCAP_PROBE_SOLARIS_SMF
 	{OVAL_SOLARIS_SMF, NULL, smf_probe_main, NULL, NULL},
 #endif
+#ifdef OPENSCAP_PROBE_SOLARIS_SMFPROPERTY
+	{OVAL_SOLARIS_SMFPROPERTY, NULL, smfproperty_probe_main, NULL, NULL},
+#endif
 #ifdef OPENSCAP_PROBE_UNIX_DNSCACHE
 	{OVAL_UNIX_DNSCACHE, NULL, dnscache_probe_main, NULL, NULL},
 #endif
diff -ruN openscap-1.3.4.org/src/OVAL/probes/unix/solaris/CMakeLists.txt openscap-1.3.4/src/OVAL/probes/unix/solaris/CMakeLists.txt
--- openscap-1.3.4.org/src/OVAL/probes/unix/solaris/CMakeLists.txt	2021-03-01 22:46:29.549752120 +0000
+++ openscap-1.3.4/src/OVAL/probes/unix/solaris/CMakeLists.txt	2021-03-01 22:47:18.170137321 +0000
@@ -11,6 +11,14 @@
 		"smf.h"
 	)
 endif()
+
+if(OPENSCAP_PROBE_SOLARIS_SMF)
+	list(APPEND SOLARIS_PROBES_SOURCES
+		"smfproperty.c"
+		"smfproperty.h"
+	)
+endif()
+
 add_library(solaris_probes_object OBJECT ${SOLARIS_PROBES_SOURCES})
 set_oscap_generic_properties(solaris_probes_object)
 target_include_directories(solaris_probes_object PRIVATE ${SOLARIS_PROBES_INCLUDE_DIRECTORIES})
diff -ruN openscap-1.3.4.org/src/OVAL/probes/unix/solaris/smfproperty.c openscap-1.3.4/src/OVAL/probes/unix/solaris/smfproperty.c
--- openscap-1.3.4.org/src/OVAL/probes/unix/solaris/smfproperty.c	1969-12-31 16:00:00.000000000 +0000
+++ openscap-1.3.4/src/OVAL/probes/unix/solaris/smfproperty.c	2021-03-01 22:47:18.170648596 +0000
@@ -0,0 +1,468 @@
+/**
+ * @file smfproperty.c
+ * @brief smfproperty probe
+ * @author "Jacob Varughese" <jacob.varughese@oracle.com>
+ *
+ * This probe processes retrieves the properties of smf services.
+ */
+
+
+#include "probe-api.h"
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#if	defined(__SVR4) && defined(__sun)
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include <ctype.h>
+#include <sys/types.h>
+#include <limits.h>
+#include <unistd.h>
+#include <libscf.h>
+#include <libscf_priv.h>
+#include "_seap.h"
+#include "probe/entcmp.h"
+#include <stdlib.h>
+#include "common/debug_priv.h"
+
+#ifndef	SCF_PG_START
+#define	SCF_PG_START	"start"
+#endif
+
+#ifndef	SCF_SNAPSHOT_RUNNING
+#define	SCF_SNAPSHOT_RUNNING	"running"
+#endif
+
+#define	MAX_SCF_VALUE_LENGTH	1024
+#define	STR(x)	((x == NULL) ? "" : x)
+/* Convenience structure for the results being reported */
+
+/*
+ * smfproperty probe:
+ *
+ *
+ * fmri
+ * service
+ * instance
+ * property
+ * value
+ */
+
+
+/* Convenience structure for the results being reported */
+struct result_info {
+	char *fmri;
+	char *service;
+	char *instance;
+	char *property;
+	char *value;
+};
+
+
+static char *
+strdup_check(char *p)
+{
+	char *ret;
+
+	if ((ret = strdup(p)) == NULL) {
+		dE("Error: Out of Memory error.\n");
+		exit(PROBE_ENOMEM);
+	}
+	return ret;
+}
+
+
+
+static void
+report_smfproperty(struct result_info *res, probe_ctx *ctx)
+{
+	SEXP_t *item;
+
+	item = probe_item_create(OVAL_SOLARIS_SMFPROPERTY, NULL,
+	    "fmri", OVAL_DATATYPE_STRING, res->fmri,
+	    "service", OVAL_DATATYPE_STRING, res->service,
+	    "instance", OVAL_DATATYPE_STRING, res->instance,
+	    "property", OVAL_DATATYPE_STRING, res->property,
+	    "value", OVAL_DATATYPE_STRING,
+	    res->value,
+	    NULL);
+	probe_item_collect(ctx, item);
+}
+
+
+static int
+convert_prop(scf_handle_t *hdl, const scf_property_t *prop,
+    struct result_info *res)
+{
+	char propname[256];
+	scf_iter_t *iter = NULL;
+	scf_value_t *value = NULL;
+	ssize_t size = 0, len = 0, i = 0;
+	char *buffer;
+	int cur_size;
+
+	if (prop == NULL)
+		return 0;
+
+	if (scf_property_get_name(prop, propname,
+	    sizeof(propname)) < 0) {
+		dE("Cannot get name of property\n");
+		dE("Error: %s\n", scf_strerror(scf_error()));
+		return 1;
+	}
+	if ((value = scf_value_create(hdl)) == NULL) {
+		dE("Cannot create value from handle for prop:%s\n",
+		    propname);
+		dE("Error: %s\n", scf_strerror(scf_error()));
+		return 1;
+	}
+	if ((iter = scf_iter_create(hdl)) == NULL) {
+		dE("Cannot create value iterator for prop:%s\n",
+		    propname);
+		dE("Error: %s\n", scf_strerror(scf_error()));
+		return 1;
+	}
+	if (scf_iter_property_values(iter, prop) != 0) {
+		dE("Cannot iterate values  for prop:%s\n",
+		    propname);
+		dE("Error: %s\n", scf_strerror(scf_error()));
+		return 1;
+	}
+	buffer = malloc(MAX_SCF_VALUE_LENGTH);
+	if (buffer == NULL) {
+		dE("Out of Memory Error.\n");
+		exit(PROBE_ENOMEM);
+	}
+	buffer[0] = '\0';
+	cur_size = MAX_SCF_VALUE_LENGTH;
+	while ((scf_iter_next_value(iter, value)) > 0) {
+		char *tmp;
+		i++;
+		size = scf_value_get_as_string(value, NULL, 0) + 1;
+		if (size < 0) {
+			dE("Cannot get value for prop:%s size is"
+			    " invalid\n", propname);
+			dE("Error: %s\n", scf_strerror(scf_error()));
+			return 1;
+		}
+		if ((size + len + 2) > cur_size) {
+			cur_size = cur_size * 2 + size;
+			buffer = realloc(buffer, cur_size);
+			if (buffer == NULL) {
+				dE("Out of Memory Error.\n");
+				exit(PROBE_ENOMEM);
+			}
+		}
+		if (i > 1) {
+			buffer[len -1] = ' ';
+			buffer[len] = '\0';
+			tmp = &buffer[len];
+		} else {
+			tmp = &buffer[0];
+		}
+		scf_value_get_as_string(value, tmp, size);
+		len += size;
+	}
+	res->value = buffer;
+	return 0;
+}
+
+static int get_smf_nested_pg(scf_handle_t *h, scf_propertygroup_t *apg, char*aprop_name, scf_propertygroup_t **appg, char**app_prop)
+{
+        char *propgrp;
+        char *prop;
+        char props[2048];
+	 static int count = 0;
+
+        props[0] = '\0';
+        snprintf(props, sizeof(props), "%s", aprop_name); 
+	dI("In get_smf_nested_pg %s\n ", props);
+	if ((prop = strstr(props, SCF_FMRI_NESTED_PROPERTYGRP_PREFIX)) != NULL) {
+		scf_propertygroup_t *pg = scf_pg_create(h);
+		*prop = '\0';
+		prop += strlen(SCF_FMRI_NESTED_PROPERTYGRP_PREFIX);
+		propgrp = props;
+		dI("property-grp:%s\tproperty:%s\n", propgrp, prop);
+		if (scf_pg_get_pg(apg, propgrp, pg) == -1 ) {
+			dE("Error: Could not read %s/%s property group \n",
+			    propgrp, prop);
+                        scf_pg_destroy(pg);
+			count = 0;
+			return 1;
+		} else {
+			count++;
+			*appg = pg;
+			*app_prop = prop;
+			 if (count > 1)
+                            scf_pg_destroy(apg);
+			return get_smf_nested_pg(h, pg, prop, appg, app_prop);
+                }
+        } else {
+                *appg = apg;
+                *app_prop = aprop_name;
+		 count = 0;
+                return 0;
+        }
+}
+
+static int
+get_smf_prop_from_pg(scf_handle_t *h, scf_propertygroup_t *pg, char *prop_name, 
+ struct result_info *res)
+{
+	scf_propertygroup_t *apg = scf_pg_create(h);
+	scf_property_t *prop = scf_property_create(h);
+	int rc=0;
+	char *aprop;
+ 
+ 	if ((rc = get_smf_nested_pg(h, pg, prop_name, &apg, &aprop)) == 0) {
+		char grp[1024];
+		dI("prop_name:%s\n", aprop);
+		if (scf_pg_get_name(apg, grp, sizeof(grp)) > 0)
+			dI("prop_grp_name:%s\n", grp);
+		if (scf_pg_get_property(apg, aprop, prop) == -1 ) {
+			dE("Error: Could not read %s property %s\n",
+		                prop_name, res->fmri);
+			rc = 1;
+			goto cleanup;
+		} else {
+			rc = convert_prop(h, prop, res);
+                }
+	}
+cleanup:
+	scf_property_destroy(prop);
+	if (apg != pg)
+		scf_pg_destroy(apg);
+	return rc;
+}
+
+static int
+get_smf_prop(scf_handle_t *h, const scf_service_t *service,
+    const scf_instance_t *instance, const char *prop_grp_name,
+    const char *prop_name, struct result_info *res, probe_ctx *ctx)
+{
+	scf_propertygroup_t *pg = scf_pg_create(h);
+	scf_property_t *prop = scf_property_create(h);
+	int rc = 0;
+
+	if (pg == NULL) {
+		dE("Unable to create property group from handle.\n");
+		dE("prop_grp_name=%s prop_name=%s.\n",
+		    prop_grp_name, prop_name);
+		dE("Error: %s.\n", scf_strerror(scf_error()));
+		rc = 1;
+		goto cleanup;
+	}
+	if (prop == NULL) {
+		dE("Unable to create property from handle.\n");
+		dE("prop_grp_name=%s prop_name=%s.\n",
+		    prop_grp_name, prop_name);
+		dE("Error: %s.\n", scf_strerror(scf_error()));
+		rc = 1;
+		goto cleanup;
+	}
+
+	if (instance == NULL) {
+		if (scf_service_get_pg(service, prop_grp_name, pg) == -1 ||
+		    scf_pg_get_property(pg, prop_name, prop) == -1) {
+			dE("Error: Could not read %s/%s from service.\n",
+			    prop_grp_name, prop_name);
+			rc = 1;
+			goto cleanup;
+		}
+	} else {
+		scf_snapshot_t *snap;
+		if ((snap = scf_snapshot_create(h)) == NULL) {
+			dE("Unable to create snapshot from handle.\n");
+			dE("fmri=%s \n", res->fmri);
+			dE("Error: %s.\n", scf_strerror(scf_error()));
+			rc = 1;
+			goto cleanup;
+		}
+		if (scf_instance_get_snapshot(instance, SCF_SNAPSHOT_RUNNING,
+		    snap) == -1) {
+			dE("Unable to get running snapshot.\n");
+			dE("for fmri=%s \n", res->fmri);
+			dE("Error: %s.\n", scf_strerror(scf_error()));
+			rc = 1;
+			goto cleanup;
+		}
+		if (scf_instance_get_pg_composed(instance, snap,
+		    prop_grp_name, pg) == -1) {
+			dE("Error: Could not read "
+			    " %s/%s property from snapshot for %s\n",
+			    prop_grp_name, prop_name, res->fmri);
+			if (scf_instance_get_pg_composed(instance, NULL,
+                            prop_grp_name, pg) == -1) {
+                                dE( "Error: Could not read "
+                                    " %s/%s property for %s\n",
+                                    prop_grp_name, prop_name, res->fmri);
+                                goto cleanup;
+                        }
+
+		}
+		scf_snapshot_destroy(snap);
+	}
+ 	if ((rc = get_smf_prop_from_pg(h, pg, prop_name, res)) != 0) {
+		if (scf_instance_get_pg_composed(instance, NULL, prop_grp_name,
+		    pg) == -1) {
+			printf(
+			    "Error: Could not read "
+			    " %s/%s property for %s\n",
+			    prop_grp_name,
+			    prop_name, res->fmri);
+			rc = 1;
+			goto cleanup;
+		}
+		rc = get_smf_prop_from_pg(h, pg, prop_name, res);
+	}
+cleanup:
+	scf_property_destroy(prop);
+	scf_pg_destroy(pg);
+	return rc;
+}
+
+static int
+collect_smfprop_info(char *asvc, char *ainst, char *aprop,
+    probe_ctx *ctx)
+{
+	struct result_info r;
+	int rc = 0;
+	char *propgrp = NULL, *prop = NULL;
+	char *tmp = NULL;
+	scf_handle_t *scf_hdl;
+	scf_instance_t *inst = NULL;
+	scf_service_t *svc = NULL;
+	char *p = NULL;
+
+	memset(&r, 0, sizeof(r));
+	r.service = asvc;
+
+	if (ainst == NULL) {
+		r.fmri = strdup_check(asvc);
+	} else { /* service name is service:instance */
+		asprintf(&r.fmri, "%s:%s", asvc, ainst);
+	}
+	if (r.fmri == NULL)
+		_exit(1);
+	if ((scf_hdl = scf_handle_create(SCF_VERSION)) == NULL ||
+	    scf_handle_bind(scf_hdl) != 0 ||
+	    (svc = scf_service_create(scf_hdl)) == NULL ||
+	    (ainst != NULL &&
+	    (inst = scf_instance_create(scf_hdl)) == NULL)) {
+		rc = PROBE_EFATAL;
+		goto cleanup;
+	}
+	if (scf_handle_decode_fmri(scf_hdl, r.fmri, NULL, svc,
+	    (ainst == NULL ? NULL : inst), NULL, NULL,
+	    SCF_DECODE_FMRI_EXACT) != 0 ||
+	    scf_error() == SCF_ERROR_NOT_FOUND) {
+		dE("scf_handle_decode_fmri failed.\n");
+		goto cleanup;
+	}
+
+	tmp = strdup_check(aprop);
+	if ((p = strstr(tmp, SCF_FMRI_PROPERTY_PREFIX)) != NULL) {
+		*p = '\0';
+		p +=  strlen(SCF_FMRI_PROPERTY_PREFIX);
+		prop = p;
+	}
+	propgrp = tmp;
+	dI("r.service_name=%s\n", r.service);
+	dI("service:%s instance:%s propgrp:%s prop:%s\n",
+	    STR(asvc), STR(ainst), STR(propgrp), STR(prop));
+	r.instance = ainst;
+	r.property = aprop;
+	if (get_smf_prop(scf_hdl, svc, inst, propgrp, prop, &r, ctx) > 0) {
+		r.value = "";
+               probe_cobj_set_flag(probe_ctx_getresult(ctx), SYSCHAR_FLAG_NOT_APPLICABLE);
+		dE("get_smf_prop failed.\n");
+	}
+	report_smfproperty(&r, ctx);
+	dI("fmri=%s service=%s instance=%s property=%s value=%s\n",
+	    STR(r.fmri), STR(r.service), STR(r.instance), STR(r.property),
+	    STR(r.value));
+        if (r.value != NULL && strlen(r.value) > 0)
+	    free(r.value);
+	dI("Service exists:%s\n", r.service);
+cleanup:
+	free(tmp);
+	free(r.fmri);
+	scf_handle_unbind(scf_hdl);
+	scf_handle_destroy(scf_hdl);
+	scf_service_destroy(svc);
+	scf_instance_destroy(inst);
+	return rc;
+}
+
+int
+smfproperty_probe_main(probe_ctx *ctx, void *arg)
+{
+	SEXP_t *probe_in, *service = NULL, *instance = NULL, *property = NULL;
+	SEXP_t *service_val = NULL, *instance_val = NULL, *property_val = NULL;
+	char *service_str = NULL, *instance_str = NULL, *property_str = NULL;
+	int rc;
+
+	probe_in = probe_ctx_getobject(ctx);
+	if (probe_in == NULL) {
+		return PROBE_ENOOBJ;
+	}
+	service = probe_obj_getent(probe_in, "service", 1);
+	if (service == NULL) {
+		dE("No service in context.\n");
+		return PROBE_ENOENT;
+	}
+	instance = probe_obj_getent(probe_in, "instance", 1);
+	property = probe_obj_getent(probe_in, "property", 1);
+	if (property == NULL) {
+		dE("No property in context.\n");
+		rc = PROBE_ENOENT;
+		goto error;
+	}
+	service_val = probe_ent_getval(service);
+	if (service_val == NULL) {
+		dE("Get service value failed.\n");
+		rc = PROBE_ENOVAL;
+		goto error;
+	}
+	service_str = SEXP_string_cstr(service_val);
+	dI("service in context: %s.\n", service_str);
+	if (instance != NULL) {
+		instance_val = probe_ent_getval(instance);
+		instance_str = SEXP_string_cstr(instance_val);
+	}
+	if (instance_str != NULL && strcmp(instance_str, "") == 0)
+		instance_str = NULL;
+	dI("instance in context: %s.\n", instance_str);
+	property_val = probe_ent_getval(property);
+	if (property_val == NULL) {
+		dE("Get property value failed.\n");
+		rc = PROBE_ENOVAL;
+		goto error;
+	}
+	property_str = SEXP_string_cstr(property_val);
+	dI("property in context: %s.\n", property_str);
+	rc = collect_smfprop_info(service_str, instance_str, property_str, ctx);
+error:
+	free(service_str);
+	free(instance_str);
+	free(property_str);
+	SEXP_free(service);
+	SEXP_free(instance);
+	SEXP_free(property);
+	SEXP_free(service_val);
+	SEXP_free(instance_val);
+	SEXP_free(property_val);
+	return rc;
+}
+#else
+
+int
+smfproperty_probe_main(probe_ctx *ctx, void *probe_arg)
+{
+	return PROBE_EOPNOTSUPP;
+}
+#endif
diff -ruN openscap-1.3.4.org/src/OVAL/probes/unix/solaris/smfproperty.h openscap-1.3.4/src/OVAL/probes/unix/solaris/smfproperty.h
--- openscap-1.3.4.org/src/OVAL/probes/unix/solaris/smfproperty.h	1969-12-31 16:00:00.000000000 +0000
+++ openscap-1.3.4/src/OVAL/probes/unix/solaris/smfproperty.h	2021-03-01 22:47:18.170959601 +0000
@@ -0,0 +1,14 @@
+/**
+ * @file smfproperty.h
+ * @brief This is a include file for smfproperty.c.
+ */
+
+
+#ifndef OPENSCAP_SMFPROPERTY_PROBE_H
+#define OPENSCAP_SMFPROPERTY_PROBE_H
+
+#include "probe-api.h"
+
+int smfproperty_probe_main(probe_ctx *ctx, void *arg);
+
+#endif 
diff -ruN openscap-1.3.4.org/src/OVAL/public/oval_types.h openscap-1.3.4/src/OVAL/public/oval_types.h
--- openscap-1.3.4.org/src/OVAL/public/oval_types.h	2021-03-01 22:46:29.578537388 +0000
+++ openscap-1.3.4/src/OVAL/public/oval_types.h	2021-03-01 22:58:28.688622497 +0000
@@ -217,6 +217,7 @@
 	OVAL_SOLARIS_PATCH54 = OVAL_FAMILY_SOLARIS + 5,
 	OVAL_SOLARIS_NDD = OVAL_FAMILY_SOLARIS + 6,
 	OVAL_SOLARIS_PACKAGECHECK = OVAL_FAMILY_SOLARIS + 7,
+	OVAL_SOLARIS_SMFPROPERTY = OVAL_FAMILY_SOLARIS + 8,
 
 /// Unix subtypes
 	OVAL_UNIX_FILE = OVAL_FAMILY_UNIX + 1,
