--- old/tests/API/probes/CMakeLists.txt	2023-12-13 12:48:29.768274170 +0100
+++ new/tests/API/probes/CMakeLists.txt	2023-12-13 12:49:34.312009959 +0100
@@ -10,14 +10,6 @@ target_include_directories(test_api_prob
 )
 add_oscap_test("test_api_probes_smoke.sh")
 
-add_oscap_test_executable(test_fsdev_is_local_fs
-	"test_fsdev_is_local_fs.c"
-)
-target_include_directories(test_fsdev_is_local_fs PUBLIC
-	"${CMAKE_SOURCE_DIR}/src/OVAL/probes"
-)
-add_oscap_test("test_fsdev_is_local_fs.sh")
-
 file(GLOB_RECURSE OVAL_RESULTS_SOURCES "${CMAKE_SOURCE_DIR}/src/OVAL/results/oval_cmp*.c")
 add_oscap_test_executable(oval_fts_list
 	"oval_fts_list.c"
