--- old/utils/oscap-run-sce-script	2023-12-13 12:57:12.936291296 +0100
+++ new/utils/oscap-run-sce-script	2023-12-13 12:58:00.408857221 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 #
 # Authors:
 #      Dominique Blaze <contact@d0m.tech>
--- old/utils/oscap-ssh	2023-12-13 12:57:12.941354661 +0100
+++ new/utils/oscap-ssh	2023-12-13 12:58:00.414025712 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
 # Copyright 2015 Martin Preisler <martin@preisler.me> 
 #
