diff -ruN openscap-1.3.4.org/utils/oscap-run-sce-script openscap-1.3.4/utils/oscap-run-sce-script
--- openscap-1.3.4.org/utils/oscap-run-sce-script	2021-03-03 21:49:30.026814138 +0000
+++ openscap-1.3.4/utils/oscap-run-sce-script	2021-03-03 22:58:38.003749750 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 #
 # Authors:
 #      Dominique Blaze <contact@d0m.tech>
diff -ruN openscap-1.3.4.org/utils/oscap-ssh openscap-1.3.4/utils/oscap-ssh
--- openscap-1.3.4.org/utils/oscap-ssh	2021-03-03 21:49:30.023089700 +0000
+++ openscap-1.3.4/utils/oscap-ssh	2021-03-03 22:58:38.004159473 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
 # Copyright 2015 Martin Preisler <martin@preisler.me> 
 #
