--- old/xsl/xccdf-report-impl.xsl	2023-12-13 13:21:25.003990520 +0100
+++ new/xsl/xccdf-report-impl.xsl	2023-12-13 13:23:30.432044240 +0100
@@ -87,15 +87,21 @@ Authors:
                             <td><xsl:value-of select="$testresult/cdf:target-facts/cdf:fact[@name = 'urn:xccdf:fact:identifier']/text()"/></td>
                         </tr>
                     </xsl:if>
-                    <xsl:if test="$testresult/cdf:benchmark">
+		    <xsl:if test="$benchmark/cdf:title">
                         <tr>
-                            <th>Benchmark URL</th>
-                            <td><xsl:value-of select="$testresult/cdf:benchmark/@href"/></td>
+			    <th>Benchmark Title</th>
+			    <td><xsl:value-of select="$benchmark/cdf:title"/></td>
                         </tr>
-                        <xsl:if test="$testresult/cdf:benchmark/@id">
+			<xsl:if test="$benchmark/cdf:version">
                             <tr>
-                                <th>Benchmark ID</th>
-                                <td><xsl:value-of select="$testresult/cdf:benchmark/@id"/></td>
+				<th>Benchmark Version</th>
+				<td><xsl:value-of select="$benchmark/cdf:version"/></td>
+                            </tr>
+                        </xsl:if>
+                        <xsl:if test="$benchmark/cdf:description">
+                            <tr>
+                                <th>Benchmark Description</th>
+                                <td><xsl:value-of select="$benchmark/cdf:description"/></td>
                             </tr>
                         </xsl:if>
                     </xsl:if>
