--- old/src/XCCDF_POLICY/check_engine_plugin.c	2023-12-13 13:26:54.721564467 +0100
+++ new/src/XCCDF_POLICY/check_engine_plugin.c	2023-12-13 13:27:44.977696111 +0100
@@ -74,6 +74,7 @@ struct check_engine_plugin_def *check_en
 	}
 
 	check_engine_plugin_entry_fn entry_fn = NULL;
+	(void) dlerror();
 	*(void **)(&entry_fn) = dlsym(ret->module_handle, STRINGIZE(OPENSCAP_CHECK_ENGINE_PLUGIN_ENTRY));
 
 	if ((error = dlerror()) != NULL) {
