diff -ruN openscap-1.3.4.org/src/OVAL/probes/SEAP/_sexp-value.h openscap-1.3.4/src/OVAL/probes/SEAP/_sexp-value.h
--- openscap-1.3.4.org/src/OVAL/probes/SEAP/_sexp-value.h	2021-03-02 04:06:49.015882794 +0000
+++ openscap-1.3.4/src/OVAL/probes/SEAP/_sexp-value.h	2021-03-02 04:21:11.340652568 +0000
@@ -94,6 +94,9 @@
         uintptr_t nxsz;
         uint16_t  real;
         uint16_t  refs;
+#if defined(__sun) && defined(__sparcv9)
+	 uint32_t pad_gcc_align_hack;
+#endif
         SEXP_t    memb[];
 };
 
diff -ruN openscap-1.3.4.org/src/OVAL/probes/SEAP/sexp-value.c openscap-1.3.4/src/OVAL/probes/SEAP/sexp-value.c
--- openscap-1.3.4.org/src/OVAL/probes/SEAP/sexp-value.c	2021-03-02 04:06:49.023768985 +0000
+++ openscap-1.3.4/src/OVAL/probes/SEAP/sexp-value.c	2021-03-02 04:21:11.341092398 +0000
@@ -107,7 +107,7 @@
         _A(sz < 16);
 
 	struct SEXP_val_lblk *lblk = oscap_aligned_malloc(
-		sizeof(uintptr_t) + (2 * sizeof(uint16_t)) + (sizeof(SEXP_t) * (1 << sz)),
+		sizeof(struct SEXP_val_lblk) + (sizeof(SEXP_t) * (1 << sz)),
 		SEXP_LBLK_ALIGN
 	);
 
