--- old/Makefile.in
+++ new/Makefile.in
@@ -210,6 +210,11 @@ $(LIBSSH_OBJS): Makefile.in config.h
 $(SSHOBJS): Makefile.in config.h
 $(SSHDOBJS): Makefile.in config.h
 
+# For chroot(), make sure all libraries are loaded.
+sshd : LDFLAGS += -Wl,-z,now
+sshd-session : LDFLAGS += -Wl,-z,now
+sshd-auth : LDFLAGS += -Wl,-z,now
+
 .c.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
--- old/configure.ac
+++ new/configure.ac
@@ -219,7 +219,6 @@ if test "$GCC" = "yes" || test "$GCC" = "egcs"; then
     if test "x$use_toolchain_hardening" = "x1"; then
 	OSSH_CHECK_CFLAG_COMPILE([-D_FORTIFY_SOURCE=2])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,relro])
-	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,now])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,noexecstack])
 	# NB. -ftrapv expects certain support functions to be present in
 	# the compiler library (libgcc or similar) to detect integer operations
