--- old/configure.ac	2020-11-19 15:09:44.733173940 +0000
+++ new/configure.ac	2020-11-19 15:10:16.204505742 +0000
@@ -171,7 +171,6 @@ if test "$GCC" = "yes" || test "$GCC" =
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,retpolineplt])
 	OSSH_CHECK_CFLAG_COMPILE([-D_FORTIFY_SOURCE=2])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,relro])
-	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,now])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,noexecstack])
 	# NB. -ftrapv expects certain support functions to be present in
 	# the compiler library (libgcc or similar) to detect integer operations
--- old/Makefile.in	2020-11-19 15:09:44.738836343 +0000
+++ new/Makefile.in	2020-11-19 15:10:16.209789541 +0000
@@ -195,6 +195,9 @@ $(srcdir)/configure: configure.ac $(srcd
 	@echo "ERROR: configure is out of date; please run ${AUTORECONF} (and configure)" 1>&2
 	@exit 1
 
+# For chroot(), make sure all libraries are loaded.
+sshd : LDFLAGS += -Wl,-z,now
+
 .c.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
