--- old/Makefile.in	2023-12-19 13:24:58.365058664 +0100
+++ new/Makefile.in	2023-12-19 13:26:20.861247462 +0100
@@ -195,6 +195,9 @@ $(LIBSSH_OBJS): Makefile.in config.h
 $(SSHOBJS): Makefile.in config.h
 $(SSHDOBJS): Makefile.in config.h
 
+# For chroot(), make sure all libraries are loaded.
+sshd : LDFLAGS += -Wl,-z,now
+
 .c.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
--- old/configure.ac	2023-12-19 13:24:58.370888050 +0100
+++ new/configure.ac	2023-12-19 13:26:20.867290964 +0100
@@ -220,7 +220,6 @@ if test "$GCC" = "yes" || test "$GCC" =
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,retpolineplt])
 	OSSH_CHECK_CFLAG_COMPILE([-D_FORTIFY_SOURCE=2])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,relro])
-	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,now])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,noexecstack])
 	# NB. -ftrapv expects certain support functions to be present in
 	# the compiler library (libgcc or similar) to detect integer operations
