--- old/configure.ac	2019-11-02 12:07:13.306921146 +0000
+++ new/configure.ac	2019-11-02 12:07:13.343777514 +0000
@@ -170,7 +170,6 @@ if test "$GCC" = "yes" || test "$GCC" =
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,retpolineplt])
 	OSSH_CHECK_CFLAG_COMPILE([-D_FORTIFY_SOURCE=2])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,relro])
-	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,now])
 	OSSH_CHECK_LDFLAG_LINK([-Wl,-z,noexecstack])
 	# NB. -ftrapv expects certain support functions to be present in
 	# the compiler library (libgcc or similar) to detect integer operations
--- old/Makefile.in	2019-11-02 12:07:13.317339230 +0000
+++ new/Makefile.in	2019-11-02 12:07:13.354472806 +0000
@@ -165,6 +165,9 @@ $(srcdir)/configure: configure.ac acloca
 	@echo "ERROR: configure is out of date; please run ${AUTORECONF} (and configure)" 1>&2
 	@exit 1
 
+# For chroot(), make sure all libraries are loaded.
+sshd : LDFLAGS += -Wl,-z,now
+
 .c.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
