--- old/openbsd-compat/port-solaris.c	2020-11-19 15:11:27.189308937 +0000
+++ new/openbsd-compat/port-solaris.c	2020-11-19 15:11:57.692303508 +0000
@@ -301,6 +301,15 @@ solaris_drop_privs_pinfo_net_fork_exec(v
 #ifdef PRIV_NET_ACCESS
 	    priv_delset(npset, PRIV_NET_ACCESS) != 0 ||
 #endif
+#ifdef PRIV_DAX_ACCESS
+	    priv_delset(npset, PRIV_DAX_ACCESS) != 0 ||
+#endif
+#ifdef PRIV_SYS_IB_INFO
+	    priv_delset(npset, PRIV_SYS_IB_INFO) != 0 ||
+#endif
+#ifdef PRIV_PROC_SELF
+	    priv_delset(npset, PRIV_PROC_SELF) != 0 ||
+#endif
 	    priv_delset(npset, PRIV_PROC_FORK) != 0 ||
 	    priv_delset(npset, PRIV_PROC_INFO) != 0 ||
 	    priv_delset(npset, PRIV_PROC_SESSION) != 0)
--- old/sandbox-solaris.c	2020-11-19 15:11:27.194959600 +0000
+++ new/sandbox-solaris.c	2020-11-19 15:11:57.697733021 +0000
@@ -68,6 +68,9 @@ ssh_sandbox_init(struct monitor *monitor
 #ifdef PRIV_SYS_IB_INFO
 	    priv_delset(box->pset, PRIV_SYS_IB_INFO) != 0 ||
 #endif
+#ifdef PRIV_PROC_SELF
+	    priv_delset(box->pset, PRIV_PROC_SELF) != 0 ||
+#endif
 	    priv_delset(box->pset, PRIV_PROC_EXEC) != 0 ||
 	    priv_delset(box->pset, PRIV_PROC_FORK) != 0 ||
 	    priv_delset(box->pset, PRIV_PROC_INFO) != 0 ||
