--- openssh-9.4p1/openbsd-compat/port-solaris.c
+++ openssh-9.4p1/openbsd-compat/port-solaris.c
@@ -300,6 +300,15 @@ if (priv_delset(npset, PRIV_PROC_EXEC) !
 #ifdef PRIV_NET_ACCESS
 	    priv_delset(npset, PRIV_NET_ACCESS) != 0 ||
 #endif
+#ifdef PRIV_DAX_ACCESS
+	    priv_delset(npset, PRIV_DAX_ACCESS) != 0 ||
+#endif
+#ifdef PRIV_SYS_IB_INFO
+	    priv_delset(npset, PRIV_SYS_IB_INFO) != 0 ||
+#endif
+#ifdef PRIV_PROC_SELF
+	    priv_delset(npset, PRIV_PROC_SELF) != 0 ||
+#endif
 	    priv_delset(npset, PRIV_PROC_FORK) != 0 ||
 	    priv_delset(npset, PRIV_PROC_INFO) != 0 ||
 	    priv_delset(npset, PRIV_PROC_SESSION) != 0)
--- openssh-9.4p1/sandbox-solaris.c
+++ openssh-9.4p1/sandbox-solaris.c
@@ -68,6 +68,9 @@ #endif
 #ifdef PRIV_SYS_IB_INFO
 	    priv_delset(box->pset, PRIV_SYS_IB_INFO) != 0 ||
 #endif
+#ifdef PRIV_PROC_SELF
+	    priv_delset(box->pset, PRIV_PROC_SELF) != 0 ||
+#endif
 	    priv_delset(box->pset, PRIV_PROC_EXEC) != 0 ||
 	    priv_delset(box->pset, PRIV_PROC_FORK) != 0 ||
 	    priv_delset(box->pset, PRIV_PROC_INFO) != 0 ||
