Origin: https://anongit.mindrot.org/openssh.git/commit/?h=V_10_2&id=434ba7684054c0637ce8f2486aaacafe65d9b8aa
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=3879

--- old/Makefile.in	2026-03-26 15:17:45.613156285 +0100
+++ new/Makefile.in	2026-03-26 15:17:46.840451275 +0100
@@ -116,7 +116,7 @@ P11OBJS= ssh-pkcs11-client.o
 SKOBJS=	ssh-sk-client.o
 
 SSHOBJS= ssh.o readconf.o clientloop.o sshtty.o \
-	sshconnect.o sshconnect2.o mux.o $(P11OBJS) $(SKOBJS)
+	sshconnect.o sshconnect2.o mux.o ssh-pkcs11.o $(SKOBJS)
 
 SSHDOBJS=sshd.o \
 	platform-listen.o \
