#
# Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
#
--- a/symbols/sun_vndr//ara	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//ara	2017-05-10 16:42:43.856608880 +0000
@@ -128,4 +130,9 @@
 partial alphanumeric_keys
 	xkb_symbols "olpc" {
 	include "ara(olpc)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "mac" {
+	include "ara(mac)"
 };
--- a/symbols/sun_vndr//ca	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//ca	2017-05-10 16:42:43.972562535 +0000
@@ -63,9 +65,9 @@
    key <AB01> { [               z, Z,               guillemotleft              ]};
    key <AB02> { [               x, X,               guillemotright             ]};
    key <AB07> { [               m, M,               mu                         ]};
-   key <AB08> { [           comma, apostrophe,      less                       ]};
+   key <AB08> { [           comma, quoteright,      less                       ]};
    key <AB09> { [          period, quotedbl,        greater                    ]};
-   key <AB10> { [          eacute, Eacute,          grave                      ]};
+   key <AB10> { [          eacute, Eacute,          quoteleft                  ]};
    
 	include "level3(ralt_switch)"
 
--- a/symbols/sun_vndr//ch	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//ch	2017-05-10 16:42:44.011177614 +0000
@@ -44,8 +46,8 @@
     key <AE07> { [          7,      slash,    brokenbar             ] };
     key <AE08> { [          8,  parenleft,         cent             ] };
     key <AE09> { [          9, parenright                           ] };
-    key <AE10> { [          0,      equal,        grave             ] };
-    key <AE11> { [ apostrophe,   question,   dead_acute             ] };
+    key <AE10> { [          0,      equal,    quoteleft             ] };
+    key <AE11> { [ quoteright,   question,   dead_acute             ] };
     key <AE12> { [ dead_circumflex, dead_grave, dead_tilde          ] };
 
     key <AD03> { [          e,          E,     EuroSign             ] };
@@ -84,8 +86,8 @@
     key <AE07> { [          7,      slash,    brokenbar             ] };
     key <AE08> { [          8,  parenleft,         cent             ] };
     key <AE09> { [          9, parenright                           ] };
-    key <AE10> { [          0,      equal,        grave             ] };
-    key <AE11> { [ apostrophe,   question,   dead_acute             ] };
+    key <AE10> { [          0,      equal,    quoteleft             ] };
+    key <AE11> { [ quoteright,   question,   dead_acute             ] };
     key <AE12> { [ dead_circumflex, dead_grave, dead_tilde          ] };
 
     key <AD03> { [          e,          E,     EuroSign             ] };
--- a/symbols/sun_vndr//cz	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//cz	2017-05-10 16:42:44.050922439 +0000
@@ -35,7 +37,7 @@
 
 	include "sun_vndr/us(sun_type6)"
 
-    key <TLDE>	{ [ semicolon, dead_abovering,	grave,          asciitilde 	]	};
+    key <TLDE>	{ [ semicolon, dead_abovering,	quoteleft,     asciitilde 	]	};
     key <AE01>	{ [      plus,          1,	exclam,     asciitilde	 	]	};
     key <AE02>	{ [    ecaron,          2,	at,     dead_caron 		]	};
     key <AE03>	{ [    scaron,          3,	numbersign,  dead_circumflex 	] 	};
@@ -108,4 +110,9 @@
 partial alphanumeric_keys
 	xkb_symbols "dvorak-ucw" {
 	include "cz(dvorak-ucw)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "rus" {
+	include "cz(rus)"
 };
--- a/symbols/sun_vndr//dk	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//dk	2017-05-10 16:42:44.129691177 +0000
@@ -85,4 +87,9 @@
 partial alphanumeric_keys
 	xkb_symbols "dvorak" {
 	include "dk(dvorak)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "winkeys" {
+	include "dk(winkeys)"
 };
--- a/symbols/sun_vndr//es	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//es	2017-05-10 16:42:44.205870643 +0000
@@ -114,4 +116,9 @@
 partial alphanumeric_keys
 	xkb_symbols "olpc2" {
 	include "es(olpc2)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "winkeys" {
+	include "es(winkeys)"
 };
--- a/symbols/sun_vndr//fi	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//fi	2017-05-10 16:42:44.243455506 +0000
@@ -52,7 +54,7 @@
     key <AD12> { [  dead_diaeresis,  asciicircum,      asciitilde             ] };
     key <AC10> { [  odiaeresis,      Odiaeresis                               ] };
     key <AC11> { [  adiaeresis,      Adiaeresis                               ] };
-    key <BKSL> { [  apostrophe,      asterisk,         grave                  ] };
+    key <BKSL> { [  apostrophe,      asterisk,         quoteleft              ] };
     key <LSGT> { [  less,            greater,          bar                    ] };
     key <AB08> { [  comma,           semicolon                                ] };
     key <AB09> { [  period,          colon                                    ] };
@@ -95,4 +97,14 @@
 partial alphanumeric_keys
 	xkb_symbols "mac" {
 	include "fi(mac)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "winkeys" {
+	include "fi(winkeys)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "das" {
+	include "fi(das)"
 };
--- a/symbols/sun_vndr//gb	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//gb	2017-05-10 16:42:44.319488626 +0000
@@ -34,7 +36,7 @@
 
 	include "sun_vndr/us(sun_type6)"
 
-    key <TLDE> { [      grave,    notsign,  brokenbar ] };
+    key <TLDE> { [  quoteleft,    notsign,  brokenbar ] };
 
     key <AE02> { [          2,   quotedbl             ] };
     key <AE03> { [          3,   sterling             ] };
--- a/symbols/sun_vndr//it	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//it	2017-05-10 16:42:44.401321565 +0000
@@ -96,4 +98,9 @@
 partial alphanumeric_keys
 	xkb_symbols "ibm" {
 	include "it(ibm)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "winkeys" {
+	include "it(winkeys)"
 };
--- a/symbols/sun_vndr//jp	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//jp	2017-05-10 16:42:44.439986096 +0000
@@ -20,14 +20,27 @@
 // FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 // NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 // WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-// 
+//
 // Except as contained in this notice, the name of a copyright holder
 // shall not be used in advertising or otherwise to promote the sale, use
 // or other dealings in this Software without prior written authorization
 // of the copyright holder.
 //
+//
+//
 
 partial default alphanumeric_keys modifier_keys
+xkb_symbols "sun_type7" {
+
+    include "jp"
+
+    key <ALGR> { [ Alt_R, Alt_R ] };
+
+	modifier_map Mod1 { Alt_L, Alt_R };
+
+};
+
+partial alphanumeric_keys modifier_keys
 xkb_symbols "sun_type6_suncompat" {
 
 	include "inet(evdev)"
@@ -57,7 +70,7 @@
     key <AD08> { [ i, I            ], [ kana_NI        ]   };
     key <AD09> { [ o, O            ], [ kana_RA        ]   };
     key <AD10> { [ p, P            ], [ kana_SE        ]   };
-    key <AD11> { [ at, grave       ], [ voicedsound    ]   };
+    key <AD11> { [ at, quoteleft   ], [ voicedsound    ]   };
     key <AD12> { [ bracketleft,    braceleft ], [ semivoicedsound, kana_openingbracket ] };
 
     key <AC01> { [ a, A            ], [ kana_CHI       ]   };
@@ -86,10 +99,13 @@
     key <AB11> { [ backslash, underscore], [ kana_RO       ]   };
     key <UNDR> { [ backslash, underscore], [ kana_RO       ]   };
 
-    key <EXEC> { [ Execute		            ]	};
-    key <KANJ> { [ Kanji		            ]	};
-    key <HENK> { [ Henkan_Mode		        ]	};
-    key <ALGR> { [ Kana_Lock, Mode_switch   ]	};
+    key <EXEC> { [ Execute                ] };
+    key <NFER> { [ Execute                ] };
+    key <KANJ> { [ Kanji                  ] };
+    key <XFER> { [ Kanji                  ] };
+    key <HENK> { [ Henkan_Mode            ] };
+    key <HKTG> { [ Henkan_Mode            ] };
+    key <ALGR> { [ Kana_Lock, Mode_switch ] };
 
 	key <COMP> { [ Multi_key				]	};
 	key <SYRQ> { [ SunSys_Req				]	};
@@ -108,7 +124,7 @@
 
 };
 
-partial alphanumeric_keys
+partial alphanumeric_keys modifier_keys
 xkb_symbols "sun_type7_suncompat" {
 
     include "sun_vndr/jp(sun_type6_suncompat)"
@@ -116,8 +132,14 @@
     key <AE10> { [ 0, asciitilde, kana_WA, kana_WO      ]   };
     key <TLDE> { [ Henkan_Mode                          ]   };
     key <HENK> { [ Kana_Lock, Mode_switch               ]   };
+    key <HKTG> { [ Kana_Lock, Mode_switch               ]   };
     key <ALGR> { [ Alt_R, Alt_R                         ]   };
+    key <XFER> { [ Kanji                                ]   };
+    key <NFER> { [ Execute                              ]   };
 
+	modifier_map Mod3 { Kana_Lock };
+	modifier_map Mod1 { Alt_L, Alt_R };
+
 };
 
 partial alphanumeric_keys
@@ -126,13 +148,6 @@
     include "sun_vndr/jp(sun_type7_suncompat)"
 };
 
-partial alphanumeric_keys
-xkb_symbols "sun_type7" {
-
-    include "jp"
-
-};
-
 partial alphanumeric_keys
 	xkb_symbols "106" {
 	include "jp(106)"
--- a/symbols/sun_vndr//kr	2015-01-15 01:13:29.000000000 +0000
+++ b/symbols/sun_vndr//kr	2017-05-10 16:42:44.478884190 +0000
@@ -58,28 +60,12 @@
 	include "kr(kr104)"
 };
 
-partial function_keys
-	xkb_symbols "hw_keys" {
-	include "kr(hw_keys)"
+partial alphanumeric_keys
+	xkb_symbols "koreankeys" {
+	include "kr(koreankeys)"
 };
 
-partial function_keys
-	xkb_symbols "ralt_hangul" {
-	include "kr(ralt_hangul)"
+partial alphanumeric_keys
+	xkb_symbols "koreankeys_raltctl" {
+	include "kr(koreankeys_raltctl)"
 };
-
-partial function_keys
-	xkb_symbols "rctrl_hangul" {
-	include "kr(rctrl_hangul)"
-};
-
-partial function_keys
-	xkb_symbols "ralt_hanja" {
-	include "kr(ralt_hanja)"
-};
-
-partial function_keys
-	xkb_symbols "rctrl_hanja" {
-	include "kr(rctrl_hanja)"
-};
-
--- a/symbols/sun_vndr//lt	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//lt	2017-05-10 16:42:44.518202232 +0000
@@ -34,7 +36,7 @@
 
 	include "sun_vndr/us(sun_type6)"
 
-    key <TLDE>  { [     grave,   asciitilde,	notsign,   brokenbar ]    };
+    key <TLDE>  { [ quoteleft,   asciitilde,	notsign,   brokenbar ]    };
     key <AE01>  { [   aogonek,      Aogonek,	1,      exclam ]    };
     key <AE02>  { [    ccaron,       Ccaron,	2,          at ]    };
     key <AE03>  { [   eogonek,      Eogonek,	3,  numbersign ]    };
--- a/symbols/sun_vndr//nl	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//nl	2017-05-10 16:42:44.654422987 +0000
@@ -44,7 +46,7 @@
     key <AE07>  { [         7, underscore,     sterling ] };
     key <AE08>  { [         8,  parenleft,    braceleft ] };
     key <AE09>  { [         9, parenright,   braceright ] };
-    key <AE10>  { [         0, apostrophe,        grave ] };
+    key <AE10>  { [         0, quoteright,   quoteleft ] };
     key <AE11>  { [     slash,   question,    backslash ] };
     key <AE12>  { [    degree, dead_tilde, dead_cedilla ] };
 
--- a/symbols/sun_vndr//no	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//no	2017-05-10 16:42:44.692920872 +0000
@@ -99,4 +101,9 @@
 partial alphanumeric_keys
 	xkb_symbols "colemak" {
 	include "no(colemak)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "winkeys" {
+	include "no(winkeys)"
 };
--- a/symbols/sun_vndr//pt	2013-09-05 22:27:17.000000000 +0000
+++ b/symbols/sun_vndr//pt	2017-05-10 16:42:44.769632754 +0000
@@ -45,7 +47,7 @@
     key <AE08> { [               8,        parenleft,     bracketleft ] };
     key <AE09> { [               9,       parenright,    bracketright ] };
     key <AE10> { [               0,            equal,      braceright ] };
-    key <AE11> { [      apostrophe,         question,           grave ] };
+    key <AE11> { [      quoteright,         question,       quoteleft ] };
     key <AE12> { [   guillemotleft,   guillemotright                  ] };
     key <AD03> { [               e,                E,        EuroSign ] };
     key <AD11> { [            plus,         asterisk,  dead_diaeresis ] }; 
--- a/symbols/sun_vndr//ru	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//ru	2017-05-10 16:42:44.846800284 +0000
@@ -201,4 +203,19 @@
 partial alphanumeric_keys
 	xkb_symbols "ruu" {
 	include "ru(ruu)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "mac" {
+	include "ru(mac)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "phonetic_azerty" {
+	include "ru(phonetic_azerty)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "phonetic_fr" {
+	include "ru(phonetic_fr)"
 };
--- a/symbols/sun_vndr//sk	2014-02-21 23:34:06.000000000 +0000
+++ b/symbols/sun_vndr//sk	2017-05-10 16:42:44.924570055 +0000
@@ -34,7 +36,7 @@
 
 	include "sun_vndr/us(sun_type6)"
 
-    key <TLDE>  { 	[       semicolon, dead_abovering, grave,             asciitilde  ]	};
+    key <TLDE>  { 	[       semicolon, dead_abovering, quoteleft,         asciitilde  ]	};
     key <AE01>  { 	[	     plus,         1,	exclam,               asciitilde  ]	};
     key <AE02>  {	[	   lcaron,         2,	at,                   dead_caron  ]	};
     key <AE03>  {	[	   scaron,         3,	numbersign,      dead_circumflex  ]	};
--- a/symbols/sun_vndr//solaris	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//solaris	2017-05-10 16:42:44.962759409 +0000
@@ -51,6 +53,8 @@
     key <LWIN> { [	Meta_L	        ] };
     key <RWIN> { [	Meta_R			] };
 
+    key <SCLK> { [	Scroll_Lock		] };
+
 // The following is needed for virtual hotkeys
 
     key <I150> { [ XF86Sleep             ] };
@@ -61,6 +65,7 @@
 
     modifier_map Shift	{ Shift_R, Shift_L };
     modifier_map Mod2	{ Num_Lock };
+    modifier_map Mod3	{ Scroll_Lock, Multi_key };
     modifier_map Mod4   { Meta_L, Meta_R };
 
 };
--- a/symbols/sun_vndr//us	2013-07-27 23:46:45.000000000 +0000
+++ b/symbols/sun_vndr//us	2017-05-10 16:42:45.117645655 +0000
@@ -26,17 +26,10 @@
 // or other dealings in this Software without prior written authorization
 // of the copyright holder.
 //
+//
+//
 
-partial default alphanumeric_keys
-xkb_symbols "sunbasic" {
-
-    include "us(basic)"
-    include "capslock(capslock)"
-    include "srvr_ctrl(xfree86)"
-
-};
-
-partial alphanumeric_keys
+default partial alphanumeric_keys
 xkb_symbols "sun_type6" {
 
 	name[Group1]= "English (US, Sun Type 6/7)";
@@ -43,12 +36,19 @@
 
     include "sun_vndr/us(sunbasic)"
     include "eurosign(4)"
-    include "level3(ralt_switch)"
 
     key <COMP> { [		Multi_key			] };
     key <SYRQ> { [		SunSys_Req			] };
 };
 
+partial alphanumeric_keys
+xkb_symbols "sunbasic" {
+
+    include "us(basic)"
+    include "capslock(capslock)"
+
+};
+
 partial alphanumeric_keys
 	xkb_symbols "basic" {
 	include "sun_vndr/us(sunbasic)"
--- a/symbols/sun_vndr/de	2014-05-06 22:47:34.000000000 +0000
+++ b/symbols/sun_vndr/de	2017-05-10 16:42:44.090146611 +0000
@@ -121,16 +123,6 @@
 };
 
 partial alphanumeric_keys
-	xkb_symbols "adnw" {
-	include "de(adnw)"
-};
-
-partial alphanumeric_keys
-	xkb_symbols "adnw_base" {
-	include "de(adnw_base)"
-};
-
-partial alphanumeric_keys
 	xkb_symbols "neo" {
 	include "de(neo)"
 };
@@ -178,4 +170,14 @@
 partial alphanumeric_keys
 	xkb_symbols "hu" {
 	include "de(hu)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "tr" {
+	include "de(tr)"
+};
+
+partial alphanumeric_keys
+	xkb_symbols "T3" {
+	include "de(T3)"
 };
