https://bugzilla.samba.org/show_bug.cgi?id=13161

--- a/lib/crypto/aes.c	2017-09-17 12:15:34.000000000 +0000
+++ b/lib/crypto/aes.c	2017-11-21 03:58:45.753212154 +0000
@@ -251,19 +251,19 @@
 void
 AES_encrypt(const unsigned char *in, unsigned char *out, const AES_KEY *key)
 {
-	if (has_intel_aes_instructions()) {
-		return AES_encrypt_aesni(in, out, key);
-	}
-	return AES_encrypt_rj(in, out, key);
+	if (has_intel_aes_instructions())
+		AES_encrypt_aesni(in, out, key);
+	else
+		AES_encrypt_rj(in, out, key);
 }
 
 void
 AES_decrypt(const unsigned char *in, unsigned char *out, const AES_KEY *key)
 {
-	if (has_intel_aes_instructions()) {
-		return AES_decrypt_aesni(in, out, key);
-	}
-	return AES_decrypt_rj(in, out, key);
+	if (has_intel_aes_instructions())
+		AES_decrypt_aesni(in, out, key);
+	else
+		AES_decrypt_rj(in, out, key);
 }
 
 #endif /* SAMBA_RIJNDAEL */
