solarisstudio 12.3 do not support this
--- a/lib/replace/wscript	2017-07-25 02:05:53.000000000 +0000
+++ b/lib/replace/wscript	2017-08-11 06:30:57.382808126 +0000
@@ -81,6 +81,8 @@
     conf.CHECK_HEADERS('sys/atomic.h')
     conf.CHECK_HEADERS('libgen.h')
 
+    if conf.CHECK_CFLAGS('-Wno-format-zero-length'):
+        conf.define('HAVE_WNO_FORMAT_ZERO_LENGTH', '1')
     if conf.CHECK_CFLAGS('-Wno-format-truncation'):
         conf.define('HAVE_WNO_FORMAT_TRUNCATION', '1')
 
@@ -714,7 +716,8 @@
                       private_library=True,
                       deps='crypt dl nsl socket rt attr' + extra_libs)
 
-    replace_test_cflags="-Wno-format-zero-length"
+    if bld.CONFIG_SET('HAVE_WNO_FORMAT_ZERO_LENGTH'):
+        replace_test_cflags="-Wno-format-zero-length"
     if bld.CONFIG_SET('HAVE_WNO_FORMAT_TRUNCATION'):
         replace_test_cflags += " -Wno-format-truncation"
     bld.SAMBA_SUBSYSTEM('replace-test',
