Fixes build, adds missing dependency on 'jansson'.

--- samba-4.18.6.orig/source3/utils/wscript_build	2023-09-12 13:06:45.460271797 +0200
+++ samba-4.18.6/source3/utils/wscript_build	2023-09-14 13:25:06.167578622 +0200
@@ -308,6 +308,7 @@ bld.SAMBA3_BINARY('destroy_netlogon_cred
                   install=False)
 
 smbstatus_source = 'status.c'
+smbstatus_extra_deps = ''
 
 if bld.CONFIG_GET("WITH_PROFILE"):
     smbstatus_source += ' status_profile.c'
@@ -316,6 +317,7 @@ else:
 
 if bld.CONFIG_GET("HAVE_JANSSON"):
     smbstatus_source += ' status_json.c'
+    smbstatus_extra_deps += 'jansson'
 else:
     smbstatus_source += ' status_json_dummy.c'
 
@@ -330,7 +332,7 @@ bld.SAMBA3_BINARY('smbstatus',
                       LOCKING
                       PROFILE
                       CONN_TDB
-                      ''')
+                      ''' + smbstatus_extra_deps)
 
 bld.SAMBA3_BINARY('mdsearch',
                  source='mdsearch.c',
