Fixes build, adds missing dependency on 'jansson'.

--- samba-4.19.8/source3/utils/wscript_build
+++ samba-4.19.8/source3/utils/wscript_build
@@ -299,6 +299,7 @@ ''',
                   install=False)
 
 smbstatus_source = 'status.c'
+smbstatus_extra_deps = ''
 
 if bld.CONFIG_GET("WITH_PROFILE"):
     smbstatus_source += ' status_profile.c'
@@ -307,6 +308,7 @@ smbstatus_source += ' status_profile_dum
 
 if bld.CONFIG_GET("HAVE_JANSSON"):
     smbstatus_source += ' status_json.c'
+    smbstatus_extra_deps += 'jansson'
 else:
     smbstatus_source += ' status_json_dummy.c'
 
@@ -321,7 +323,7 @@ smbd_base
                       LOCKING
                       PROFILE
                       CONN_TDB
-                      ''')
+                      ''' + smbstatus_extra_deps)
 
 bld.SAMBA3_BINARY('mdsearch',
                  source='mdsearch.c',
