nspr.pc for Solaris.
Will not send upstream.

--- a/nspr/config/nspr.pc.in	2017-08-02 09:57:57.750797488 +0000
+++ b/nspr/config/nspr.pc.in	2017-08-02 10:00:07.300087379 +0000

@@ -6,5 +6,5 @@
 Name: NSPR
 Description: The Netscape Portable Runtime
 Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
-Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
+Libs: -R@libdir@ -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ -lpthread -ldl -lposix4
 Cflags: -I@includedir@
