Solaris specific documentation changes.

--- unbound-1.24.1/doc/unbound.8.in
+++ unbound-1.24.1/doc/unbound.8.in
@@ -50,6 +50,14 @@ (the answer\(aqs time\-to\-live or TTL).
 A second query for the same name will then be answered from the cache.
 Unbound can also do DNSSEC validation.
 .sp
+Unbound can be enabled by authorized user within the service management
+facility (SMF), using  the  following  fault  management  resource
+identifier (FMRI):
+.sp
+.RS 6n
+svc:/network/dns/server:unbound
+.RE
+.sp
 To use a locally running Unbound for resolving put:
 .INDENT 0.0
 .INDENT 3.5
