Remove erroneous date-stamping from the egg installation.


--- setuptools-39.2.0/setup.cfg.orig	2018-06-21 16:10:12.576277447 -0700
+++ setuptools-39.2.0/setup.cfg	2018-06-21 16:10:18.571522072 -0700
@@ -1,14 +1,10 @@
 [bumpversion]
 current_version = 39.2.0
 commit = True
 tag = True
 
-[egg_info]
-tag_build = .post
-tag_date = 1
-
 [aliases]
 clean_egg_info = egg_info -Db ''
 release = clean_egg_info sdist bdist_wheel
 source = register sdist binary
 binary = bdist_egg upload --show-response
