Existing set of patches from the X gate

--- a/man/xcb-examples.man	Thu Feb 19 12:55:17 2015
+++ b/man/xcb-examples.man	Thu Feb 19 12:57:50 2015
@@ -1,4 +1,4 @@
-.TH xcb-examples __libmansuffix__ __xorgversion__ "XCB examples"
+.TH xcb-examples 3xcb "libxcb 1.11" "X Version 11" "XCB examples"
 .ad l
 .SH NAME
 xcb-examples \- manpage examples
@@ -52,8 +52,8 @@
 No line is longer than 80 characters (including indentation).
 
 .SH SEE ALSO
-.BR xcb_connect (__libmansuffix__),
-.BR xcb_get_property (__libmansuffix__),
-.BR xcb_flush (__libmansuffix__)
+.BR xcb_connect (3xcb),
+.BR xcb_get_property (3xcb),
+.BR xcb_flush (3xcb)
 .SH AUTHOR
 Michael Stapelberg <michael+xcb at stapelberg dot de>
--- a/man/xcb-requests.man	Thu Feb 19 12:55:20 2015
+++ b/man/xcb-requests.man	Thu Feb 19 12:58:51 2015
@@ -1,4 +1,4 @@
-.TH xcb-requests __libmansuffix__ __xorgversion__ "XCB examples"
+.TH xcb-requests 3xcb "libxcb 1.11" "X Version 11" "XCB examples"
 .ad l
 .SH NAME
 xcb-requests \- about request manpages
@@ -156,10 +156,10 @@
 .fi
 
 .SH SEE ALSO
-.BR xcb_map_window (__libmansuffix__),
-.BR xcb_intern_atom (__libmansuffix__),
-.BR xcb_list_fonts (__libmansuffix__),
-.BR xcb_poll_for_event (__libmansuffix__),
-.BR xcb_request_check (__libmansuffix__)
+.BR xcb_map_window (3xcb),
+.BR xcb_intern_atom (3xcb),
+.BR xcb_list_fonts (3xcb),
+.BR xcb_poll_for_event (3xcb),
+.BR xcb_request_check (3xcb)
 .SH AUTHOR
 Michael Stapelberg <michael+xcb at stapelberg dot de>
