Existing set of patches from the X gate

--- a/src/c_client.py	2016-05-09 09:10:55.000000000 -0700
+++ b/src/c_client.py	2017-02-01 08:17:22.992353664 -0800
@@ -1934,10 +1934,10 @@
 		    # compute the length of it by subtracting it from
                     # the adress of the first byte after the end of the
                     # request
-		    after_end_of_request = '(((char*)R) + R->length * 4)'
-		    start_of_list = '%s(R)' % (field.c_accessor_name)
+                    after_end_of_request = '(((char*)R) + R->length * 4)'
+                    start_of_list = '%s(R)' % (field.c_accessor_name)
                     bytesize_of_list = '%s - (char*)(%s)' % (after_end_of_request, start_of_list)
-		    return '(%s) / sizeof(%s)' % (bytesize_of_list, field.type.member.c_wiretype)
+                    return '(%s) / sizeof(%s)' % (bytesize_of_list, field.type.member.c_wiretype)
             else:
                 raise Exception(
                     "lengthless lists with varsized members are not supported. Fieldname '%s'"
--- a/src/c_client.py	2017-09-20 13:30:07.638391636 -0700
+++ b/src/c_client.py	2017-09-20 13:30:16.076266818 -0700
@@ -1932,7 +1932,7 @@
                 else:
 		    # use the accessor to get the start of the list, then
 		    # compute the length of it by subtracting it from
-                    # the adress of the first byte after the end of the
+                    # the address of the first byte after the end of the
                     # request
                     after_end_of_request = '(((char*)R) + R->length * 4)'
                     start_of_list = '%s(R)' % (field.c_accessor_name)
