https://unicode-org.atlassian.net/browse/ICU-20853
diff -r 32899d06b1e2 icu/source/common/ubidiimp.h
--- a/icu/source/common/ubidiimp.h	Tue Sep 10 23:15:21 2019 +0200
+++ b/icu/source/common/ubidiimp.h	Tue Sep 10 23:17:56 2019 +0200
@@ -24,6 +24,14 @@
 #include "unicode/uchar.h"
 #include "ubidi_props.h"
 
+/* Solaris defines CS and ES in sys/regset.h */
+#ifdef ES
+#undef ES
+#endif
+#ifdef CS
+#undef CS
+#endif
+
 /* miscellaneous definitions ---------------------------------------------- */
 
 typedef uint8_t DirProp;
