logged as https://unicode-org.atlassian.net/browse/ICU-20852

--- icu4c-65_1/icu/source/data/Makefile.in.old	2019-10-04 10:50:00.965892341 +0200
+++ icu4c-65_1/icu/source/data/Makefile.in	2019-10-04 10:50:09.265554606 +0200
@@ -148,7 +148,8 @@
 # and convert it to the current type.
 ifneq ($(ICUDATA_ARCHIVE),)
 ICUDATA_SOURCE_ARCHIVE = $(OUTDIR)/$(ICUDATA_PLATFORM_NAME).dat
-$(ICUDATA_SOURCE_ARCHIVE): $(ICUDATA_ARCHIVE) $(OUTDIR)
+$(ICUDATA_SOURCE_ARCHIVE): $(ICUDATA_ARCHIVE)
+	$(MKINSTALLDIRS) $(OUTDIR)
 	$(INVOKE) $(TOOLBINDIR)/icupkg -t$(ICUDATA_CHAR) $(ICUDATA_ARCHIVE) $(ICUDATA_SOURCE_ARCHIVE)
 endif
 else
