https://gitlab.com/wireshark/wireshark/-/issues/18493

--- wireshark-4.0.0/wsutil/CMakeLists.txt
+++ wireshark-4.0.0/wsutil/CMakeLists.txt
@@ -7,8 +7,8 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 #
 
-add_definitions(-DPLUGIN_DIR=\"${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_LIBDIR}\")
-add_definitions(-DEXTCAP_DIR=\"${CMAKE_INSTALL_PREFIX}/${EXTCAP_INSTALL_LIBDIR}\")
+add_definitions(-DPLUGIN_DIR=\"${PLUGIN_INSTALL_LIBDIR}\")
+add_definitions(-DEXTCAP_DIR=\"${EXTCAP_INSTALL_LIBDIR}\")
 add_definitions(-DDATA_DIR=\"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}\")
 
 add_subdirectory(wmem)
