We have libgweather 4 and geocode-glib built with libsoup-3

--- a/meson.build	2024-01-21 13:30:52.356005685 -0800
+++ b/meson.build	2024-01-21 13:31:13.868591244 -0800
@@ -15,9 +15,9 @@
 gobject = dependency('gobject-2.0', version: '>= 2.58')
 gtk = dependency('gtk+-3.0', version: '>=3.20')
 gsound = dependency('gsound', version: '>=0.98')
-gweather = dependency('gweather-3.0', version: '>=3.32.0')
+gweather = dependency('gweather4')
 gnomedesktop = dependency('gnome-desktop-3.0', version: '>=3.8')
-geocodeglib = dependency('geocode-glib-1.0', version: '>=1.0')
+geocodeglib = dependency('geocode-glib-2.0', version: '>=1.0')
 libgeoclue = dependency('libgeoclue-2.0', version: '>=2.4')
 libhandy = dependency ('libhandy-1', version: '>= 1.0.0', required: false)
 
