Disables GCC 7 format-nonliteral error.

--- a/vinagre/vinagre-utils.c
+++ b/vinagre/vinagre-utils.c
@@ -683,7 +683,9 @@
 	_tmp9_ = auth_label;
 	_tmp10_ = _ ("%s authentication is required");
 	_tmp11_ = protocol;
+#pragma GCC diagnostic warning "-Wformat-nonliteral"
 	_tmp12_ = g_strdup_printf (_tmp10_, _tmp11_);
+#pragma GCC diagnostic pop
 	_tmp13_ = _tmp12_;
 	gtk_label_set_label (_tmp9_, _tmp13_);
 	_g_free0 (_tmp13_);
