From a58a535028882a92b76f3814f70803872ad5727e Mon Sep 17 00:00:00 2001
From: Yann Weber <yweber@entrouvert.com>
Date: Mon, 27 Jan 2025 12:30:34 +0100
Subject: [PATCH] registry: fix lasso_registry_destroy private declaration
 (#101250)

---
 lasso/registry-private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lasso/registry-private.h b/lasso/registry-private.h
index 413c1ad4..4e0cb9f9 100644
--- a/lasso/registry-private.h
+++ b/lasso/registry-private.h
@@ -39,7 +39,7 @@ struct _LassoRegistry {
 };
 
 LassoRegistry *lasso_registry_new();
-void lasso_registry_destroy();
+void lasso_registry_destroy(LassoRegistry *registry);
 void lasso_registry_default_shutdown();
 gint lasso_registry_add_direct_mapping(LassoRegistry *registry, const char *from_namespace,
 		const char *from_name, const char *to_namespace, const char *to_name);
