--- net-snmp-5.7.3/perl/OID/OID.xs
+++ net-snmp-5.7.3/perl/OID/OID.xs
@@ -7,6 +7,8 @@
 #include "perl.h"
 #include "XSUB.h"
 
+#define INT64_INCLUDED 1
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 
--- net-snmp-5.7.3/perl/agent/agent.xs
+++ net-snmp-5.7.3/perl/agent/agent.xs
@@ -9,6 +9,8 @@
 #include <netdb.h>
 #include <sys/socket.h>
 
+#define INT64_INCLUDED 1
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <net-snmp/agent/net-snmp-agent-includes.h>
--- net-snmp-5.7.3/perl/SNMP/SNMP.xs
+++ net-snmp-5.7.3/perl/SNMP/SNMP.xs
@@ -16,6 +16,8 @@
 #include "perl.h"
 #include "XSUB.h"
 
+#define INT64_INCLUDED 1
+
 #include <net-snmp/net-snmp-config.h>
 #include <net-snmp/net-snmp-includes.h>
 #include <sys/types.h>
--- net-snmp-5.7.3/perl/TrapReceiver/TrapReceiver.xs
+++ net-snmp-5.7.3/perl/TrapReceiver/TrapReceiver.xs
@@ -7,6 +7,8 @@
 #include "perl.h"
 #include "XSUB.h"
 
+#define INT64_INCLUDED 1
+
 #include "ppport.h"
 
 #include <net-snmp/net-snmp-config.h>
