https://bugzilla.mozilla.org/show_bug.cgi?id=1737293

--- thunderbird-91.2.0/comm/mailnews/imap/src/nsImapProtocol.h
+++ thunderbird-91.2.0/comm/mailnews/imap/src/nsImapProtocol.h
@@ -56,7 +56,7 @@
 class nsIMAPMessagePartID;
 class nsIPrefBranch;
 
-#define kDownLoadCacheSize 16000u  // was 1536 - try making it bigger
+#define kDownLoadCacheSize 80000u  // was 1536 - try making it bigger
 
 typedef struct _msg_line_info {
   const char* adoptedMessageLine;
