https://bugzilla.mozilla.org/show_bug.cgi?id=1568458

--- a/comm/mail/installer/package-manifest.in
+++ b/comm/mail/installer/package-manifest.in
@@ -478,6 +478,9 @@
 #ifndef MOZ_SYSTEM_NSS
 #if defined(XP_LINUX) && !defined(ANDROID)
 @BINPATH@/@DLL_PREFIX@freeblpriv3@DLL_SUFFIX@
+#elif defined(XP_SOLARIS) && defined(SPARC64)
+@BINPATH@/@DLL_PREFIX@freebl_64fpu_3@DLL_SUFFIX@
+@BINPATH@/@DLL_PREFIX@freebl_64int_3@DLL_SUFFIX@
 #else
 @BINPATH@/@DLL_PREFIX@freebl3@DLL_SUFFIX@
 #endif
@@ -503,13 +506,6 @@
 #endif
 #endif
 
-; for Solaris SPARC
-#ifdef SOLARIS
-@BINPATH@/@DLL_PREFIX@freebl_32fpu_3@DLL_SUFFIX@
-@BINPATH@/@DLL_PREFIX@freebl_32int_3@DLL_SUFFIX@
-@BINPATH@/@DLL_PREFIX@freebl_32int64_3@DLL_SUFFIX@
-#endif
-
 ; [Updater]
 ;
 #ifdef MOZ_UPDATER
