https://bugzilla.mozilla.org/show_bug.cgi?id=1737293

--- thunderbird-153.0/comm/mailnews/imap/src/nsImapProtocol.h
+++ thunderbird-153.0/comm/mailnews/imap/src/nsImapProtocol.h
@@ -55,7 +55,7 @@
 class nsIMAPMessagePartID;
 class nsIPrefBranch;
 
-#define kDownLoadCacheSize 16000u  // was 1536 - try making it bigger
+#define kDownLoadCacheSize 80000u  // was 1536 - try making it bigger
 
 using msg_line_info = struct _msg_line_info {
   const char* adoptedMessageLine;
