https://github.com/libffi/libffi/issues/778

--- libffi-3.4.4/src/sparc/ffi64.c
+++ libffi-3.4.4/src/sparc/ffi64.c
@@ -382,13 +382,19 @@
 	  *argp++ = *(SINT32 *)a;
 	  break;
 	case FFI_TYPE_UINT32:
-	case FFI_TYPE_FLOAT:
 	  *argp++ = *(UINT32 *)a;
 	  break;
 	case FFI_TYPE_SINT64:
 	case FFI_TYPE_UINT64:
 	case FFI_TYPE_POINTER:
+	  *argp++ = *(UINT64 *)a;
+	  break;
+	case FFI_TYPE_FLOAT:
+	  flags |= SPARC_FLAG_FP_ARGS;
+	  *argp++ = *(UINT32 *)a;
+	  break;
 	case FFI_TYPE_DOUBLE:
+	  flags |= SPARC_FLAG_FP_ARGS;
 	  *argp++ = *(UINT64 *)a;
 	  break;
 
