# Solaris-specific patch.  Not appropriate for upstream.

diff --git a/gdb/Makefile.in b/gdb/Makefile.in
index c3e074b..62e2e8e 100644
--- a/gdb/Makefile.in
+++ b/gdb/Makefile.in
@@ -177,7 +177,7 @@ LIBDECNUMBER_CFLAGS = -I$(LIBDECNUMBER_DIR) -I$(LIBDECNUMBER_SRC)
 # Where is the READLINE library?  Typically in ../readline/readline.
 READLINE_DIR = ../readline/readline
 READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE = @READLINE@
+READLINE = $(READLINE_DIR)/libreadline.a $(READLINE_DIR)/libhistory.a
 READLINE_DEPS = @READLINE_DEPS@
 READLINE_CFLAGS = @READLINE_CFLAGS@
 
@@ -213,7 +213,7 @@ GDB_WERROR_CFLAGS = $(WERROR_CFLAGS)
 PTHREAD_CFLAGS = @PTHREAD_CFLAGS@
 PTHREAD_LIBS = @PTHREAD_LIBS@
 
-RDYNAMIC = @RDYNAMIC@
+RDYNAMIC =
 
 # Where is the INTL library?  Typically in ../intl.
 INTL = @LIBINTL@
@@ -539,7 +539,7 @@ SUBDIR_GDBTK_CFLAGS = -DGDBTK
 CONFIG_OBS = @CONFIG_OBS@
 CONFIG_SRCS = @CONFIG_SRCS@
 CONFIG_DEPS = @CONFIG_DEPS@
-CONFIG_LDFLAGS = @CONFIG_LDFLAGS@
+CONFIG_LDFLAGS =
 ENABLE_CFLAGS = @ENABLE_CFLAGS@
 CONFIG_ALL = @CONFIG_ALL@
 CONFIG_CLEAN = @CONFIG_CLEAN@
@@ -1606,7 +1606,7 @@ COMMON_OBS = $(DEPFILES) $(CONFIG_OBS) $(YYOBJ) \
 	$(SUBDIR_TARGET_OBS) \
 	$(SUBDIR_GCC_COMPILE_OBS)
 
-SUBDIRS = doc @subdirs@ data-directory
+SUBDIRS = @subdirs@ data-directory
 CLEANDIRS = $(SUBDIRS)
 
 # List of subdirectories in the build tree that must exist.
