diff -ur gdb-8.3.1.orig/gdb/procfs.c gdb-8.3.1/gdb/procfs.c
--- gdb-8.3.1.orig/gdb/procfs.c	2020-02-18 14:07:31.550044666 +0000
+++ gdb-8.3.1/gdb/procfs.c	2020-02-18 14:13:35.891996056 +0000
@@ -1636,7 +1636,7 @@
       return NULL;
     }
   /* Now extract the GS register's lower 16 bits.  */
-  key = (*gregs)[GS] & 0xffff;
+  key = (*gregs)[REG_GS] & 0xffff;
 
   /* Find the matching entry and return it.  */
   return proc_get_LDT_entry (pi, key);
