https://sourceware.org/bugzilla/show_bug.cgi?id=32172

--- gdb-15.1/libiberty/config.in
+++ gdb-15.1/libiberty/config.in
@@ -442,7 +442,7 @@ /* Define to 1 if `vfork' works. */
 #undef HAVE_WORKING_VFORK
 
 /* Define if you have x86 SHA1 HW acceleration support. */
-#undef HAVE_X86_SHA1_HW_SUPPORT
+/* #undef HAVE_X86_SHA1_HW_SUPPORT */
 
 /* Define to 1 if you have the `_doprnt' function. */
 #undef HAVE__DOPRNT
