GCC 10 defaults to -fno-common.

--- mysql-5.7.31/plugin/innodb_memcached/innodb_memcache/include/innodb_cb_api.h
+++ mysql-5.7.31/plugin/innodb_memcached/innodb_memcache/include/innodb_cb_api.h
@@ -394,56 +394,56 @@
 /*==================*/
 	ib_crsr_t	ib_crsr);
 
-cb_open_table_t			ib_cb_open_table;
-cb_read_row_t			ib_cb_read_row;
-cb_insert_row_t			ib_cb_insert_row;
-cb_cursor_delete_row_t		ib_cb_delete_row;
-cb_cursor_update_row_t		ib_cb_update_row;
-cb_cursor_moveto_t		ib_cb_moveto;
-cb_sec_search_tuple_create_t	ib_cb_search_tuple_create;
-cb_sec_read_tuple_create_t	ib_cb_read_tuple_create;
-cb_tuple_delete_t		ib_cb_tuple_delete;
-cb_tuple_read_u8_t		ib_cb_tuple_read_u8;
-cb_tuple_read_u16_t		ib_cb_tuple_read_u16;
-cb_tuple_read_u32_t		ib_cb_tuple_read_u32;
-cb_tuple_read_u64_t		ib_cb_tuple_read_u64;
-cb_tuple_read_i8_t		ib_cb_tuple_read_i8;
-cb_tuple_read_i16_t		ib_cb_tuple_read_i16;
-cb_tuple_read_i32_t		ib_cb_tuple_read_i32;
-cb_tuple_read_i64_t		ib_cb_tuple_read_i64;
-cb_col_set_value_t		ib_cb_col_set_value;
-cb_col_get_value_t		ib_cb_col_get_value;
-cb_col_get_meta_t		ib_cb_col_get_meta;
-cb_trx_begin_t			ib_cb_trx_begin;
-cb_trx_commit_t			ib_cb_trx_commit;
-cb_trx_rollback_t		ib_cb_trx_rollback;
-cb_trx_start_t			ib_cb_trx_start;
-cb_trx_release_t		ib_cb_trx_release;
-cb_tuple_get_n_cols_t		ib_cb_tuple_get_n_cols;
-cb_cursor_set_match_mode_t	ib_cb_cursor_set_match_mode;
-cb_cursor_lock_t		ib_cb_cursor_lock;
-cb_cursor_close_t		ib_cb_cursor_close;
-cb_cursor_new_trx_t		ib_cb_cursor_new_trx;
-cb_cursor_reset_t		ib_cb_cursor_reset;
-cb_col_get_name_t		ib_cb_col_get_name;
-cb_get_idx_field_name		ib_cb_get_idx_field_name;
-cb_table_truncate_t		ib_cb_table_truncate;
-cb_cursor_first_t		ib_cb_cursor_first;
-cb_cursor_next_t		ib_cb_cursor_next;
-cb_cursor_open_index_using_name_t	ib_cb_cursor_open_index_using_name;
-cb_close_thd_t			ib_cb_close_thd;
-cb_get_cfg_t			ib_cb_get_cfg;
-cb_cursor_set_memcached_sync	ib_cb_cursor_set_memcached_sync;
-cb_cursor_set_cluster_access_t	ib_cb_cursor_set_cluster_access;
-cb_cursor_commit_trx_t		ib_cb_cursor_commit_trx;
-cb_cfg_trx_level_t		ib_cb_cfg_trx_level;
-cb_get_n_user_cols		ib_cb_get_n_user_cols;
-cb_cursor_set_lock_t		ib_cb_cursor_set_lock;
-cb_trx_get_start_time		ib_cb_trx_get_start_time;
-cb_bk_commit_interval		ib_cb_cfg_bk_commit_interval;
-cb_ut_strerr			ib_cb_ut_strerr;
-cb_cursor_stmt_begin		ib_cb_cursor_stmt_begin;
-cb_trx_read_only_t		ib_cb_trx_read_only;
-cb_is_virtual_table		ib_cb_is_virtual_table;
+extern cb_open_table_t			ib_cb_open_table;
+extern cb_read_row_t			ib_cb_read_row;
+extern cb_insert_row_t			ib_cb_insert_row;
+extern cb_cursor_delete_row_t		ib_cb_nelete_row;
+extern cb_cursor_update_row_t		ib_cb_update_row;
+extern cb_cursor_moveto_t		ib_cb_moveto;
+extern cb_sec_search_tuple_create_t	ib_cb_search_tuple_create;
+extern cb_sec_read_tuple_create_t	ib_cb_read_tuple_create;
+extern cb_tuple_delete_t		ib_cb_tuple_delete;
+extern cb_tuple_read_u8_t		ib_cb_tuple_read_u8;
+extern cb_tuple_read_u16_t		ib_cb_tuple_read_u16;
+extern cb_tuple_read_u32_t		ib_cb_tuple_read_u32;
+extern cb_tuple_read_u64_t		ib_cb_tuple_read_u64;
+extern cb_tuple_read_i8_t		ib_cb_tuple_read_i8;
+extern cb_tuple_read_i16_t		ib_cb_tuple_read_i16;
+extern cb_tuple_read_i32_t		ib_cb_tuple_read_i32;
+extern cb_tuple_read_i64_t		ib_cb_tuple_read_i64;
+extern cb_col_set_value_t		ib_cb_col_set_value;
+extern cb_col_get_value_t		ib_cb_col_get_value;
+extern cb_col_get_meta_t		ib_cb_col_get_meta;
+extern cb_trx_begin_t			ib_cb_trx_begin;
+extern cb_trx_commit_t			ib_cb_trx_commit;
+extern cb_trx_rollback_t		ib_cb_trx_rollback;
+extern cb_trx_start_t			ib_cb_trx_start;
+extern cb_trx_release_t		ib_cb_trx_release;
+extern cb_tuple_get_n_cols_t		ib_cb_tuple_get_n_cols;
+extern cb_cursor_set_match_mode_t	ib_cb_cursor_set_match_mode;
+extern cb_cursor_lock_t		ib_cb_cursor_lock;
+extern cb_cursor_close_t		ib_cb_cursor_close;
+extern cb_cursor_new_trx_t		ib_cb_cursor_new_trx;
+extern cb_cursor_reset_t		ib_cb_cursor_reset;
+extern cb_col_get_name_t		ib_cb_col_get_name;
+extern cb_get_idx_field_name		ib_cb_get_idx_field_name;
+extern cb_table_truncate_t		ib_cb_table_truncate;
+extern cb_cursor_first_t		ib_cb_cursor_first;
+extern cb_cursor_next_t		ib_cb_cursor_next;
+extern cb_cursor_open_index_using_name_t	ib_cb_cursor_open_index_using_name;
+extern cb_close_thd_t			ib_cb_close_thd;
+extern cb_get_cfg_t			ib_cb_get_cfg;
+extern cb_cursor_set_memcached_sync	ib_cb_cursor_set_memcached_sync;
+extern cb_cursor_set_cluster_access_t	ib_cb_cursor_set_cluster_access;
+extern cb_cursor_commit_trx_t		ib_cb_cursor_commit_trx;
+extern cb_cfg_trx_level_t		ib_cb_cfg_trx_level;
+extern cb_get_n_user_cols		ib_cb_get_n_user_cols;
+extern cb_cursor_set_lock_t		ib_cb_cursor_set_lock;
+extern cb_trx_get_start_time		ib_cb_trx_get_start_time;
+extern cb_bk_commit_interval		ib_cb_cfg_bk_commit_interval;
+extern cb_ut_strerr			ib_cb_ut_strerr;
+extern cb_cursor_stmt_begin		ib_cb_cursor_stmt_begin;
+extern cb_trx_read_only_t		ib_cb_trx_read_only;
+extern cb_is_virtual_table		ib_cb_is_virtual_table;
 
 #endif /* innodb_cb_api_h */
--- mysql-5.7.31/plugin/innodb_memcached/innodb_memcache/src/innodb_api.c
+++ mysql-5.7.31/plugin/innodb_memcached/innodb_memcache/src/innodb_api.c
@@ -44,6 +44,52 @@
 /** Whether to update all columns' value or a specific column value */
 #define UPDATE_ALL_VAL_COL	-1
 
+cb_open_table_t                  ib_cb_open_table;
+cb_read_row_t                    ib_cb_read_row;
+cb_insert_row_t                 ib_cb_insert_row;
+cb_cursor_delete_row_t          ib_cb_delete_row;
+cb_cursor_update_row_t          ib_cb_update_row;
+cb_cursor_moveto_t              ib_cb_moveto;
+cb_sec_search_tuple_create_t    ib_cb_search_tuple_create;
+cb_sec_read_tuple_create_t      ib_cb_read_tuple_create;
+cb_tuple_delete_t               ib_cb_tuple_delete;
+cb_tuple_read_u8_t              ib_cb_tuple_read_u8;
+cb_tuple_read_u16_t             ib_cb_tuple_read_u16;
+cb_tuple_read_u32_t             ib_cb_tuple_read_u32;
+cb_tuple_read_u64_t             ib_cb_tuple_read_u64;
+cb_tuple_read_i8_t              ib_cb_tuple_read_i8;
+cb_tuple_read_i16_t             ib_cb_tuple_read_i16;
+cb_tuple_read_i32_t             ib_cb_tuple_read_i32;
+cb_tuple_read_i64_t             ib_cb_tuple_read_i64;
+cb_col_set_value_t              ib_cb_col_set_value;
+cb_col_get_value_t              ib_cb_col_get_value;
+cb_col_get_meta_t               ib_cb_col_get_meta;
+cb_trx_begin_t                  ib_cb_trx_begin;
+cb_trx_commit_t                 ib_cb_trx_commit;
+cb_trx_rollback_t               ib_cb_trx_rollback;
+cb_trx_start_t                  ib_cb_trx_start;
+cb_trx_release_t                ib_cb_trx_release;
+cb_tuple_get_n_cols_t           ib_cb_tuple_get_n_cols;
+cb_cursor_set_match_mode_t      ib_cb_cursor_set_match_mode;
+cb_cursor_lock_t                ib_cb_cursor_lock;
+cb_cursor_close_t               ib_cb_cursor_close;
+cb_cursor_new_trx_t             ib_cb_cursor_new_trx;
+cb_cursor_reset_t               ib_cb_cursor_reset;
+cb_col_get_name_t               ib_cb_col_get_name;
+cb_get_idx_field_name           ib_cb_get_idx_field_name;
+cb_table_truncate_t             ib_cb_table_truncate;
+cb_cursor_first_t               ib_cb_cursor_first;
+cb_cursor_next_t                ib_cb_cursor_next;
+cb_cfg_trx_level_t              ib_cb_cfg_trx_level;
+cb_get_n_user_cols               ib_cb_get_n_user_cols;
+cb_cursor_set_lock_t             ib_cb_cursor_set_lock;
+cb_trx_get_start_time            ib_cb_trx_get_start_time;
+cb_bk_commit_interval            ib_cb_cfg_bk_commit_interval;
+cb_ut_strerr                     ib_cb_ut_strerr;
+cb_cursor_stmt_begin             ib_cb_cursor_stmt_begin;
+cb_trx_read_only_t               ib_cb_trx_read_only;
+cb_is_virtual_table              ib_cb_is_virtual_table;
+
 extern bool    release_mdl_lock;
 
 extern option_t config_option_names[];
