From 76e51a6343000165747b39228c0070bbbfa346ba Mon Sep 17 00:00:00 2001
From: Arvind Umrao <arvind.umrao@oracle.com>
Date: Tue, 8 May 2012 11:58:06 +0530
Subject: [PATCH] 7162172

---
 Xtranssock.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git Xtranssock.c Xtranssock.c
--- a/Xtranssock.c	2023-09-19 13:34:19.760227324 -0700
+++ b/Xtranssock.c	2023-09-19 13:45:07.489738999 -0700
@@ -1639,7 +1639,7 @@
 	 * only affect one of a set of addresses.
 	 */
 
-	if (olderrno == ECONNREFUSED || olderrno == EINTR
+	if (olderrno == ECONNREFUSED
 #if defined(IPv6) && defined(AF_INET6)
 	  || (((addrlist->addr->ai_next != NULL) ||
 	        (addrlist->addr != addrlist->firstaddr)) &&
@@ -1652,7 +1652,7 @@
 #endif
 	    )
 	    res = TRANS_TRY_CONNECT_AGAIN;
-	else if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS)
+	else if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS || olderrno == EINTR)
 	    res = TRANS_IN_PROGRESS;
 	else
 	{
@@ -1925,7 +1925,7 @@
 	    if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS)
 		return TRANS_IN_PROGRESS;
 	    else if (olderrno == EINTR)
-		return TRANS_TRY_CONNECT_AGAIN;
+		return TRANS_IN_PROGRESS;
 	    else {
 		prmsg (2,"SocketUNIXConnect: Can't connect: errno = %d\n",
 		       EGET());
