link the introspection lib with ssp

--- a/src/meson.build	2020-06-07 10:39:48.170836333 -0700
+++ b/src/meson.build	2020-06-07 10:40:41.333403389 -0700
@@ -131,7 +131,7 @@
         install_dir_gir: girdir,
     install_dir_typelib: typelibdir,
         export_packages: libdazzle_package,
-             extra_args: [ '--c-include=dazzle.h', '--quiet', '-DDAZZLE_COMPILATION', ],
+             extra_args: [ '--c-include=dazzle.h', '--quiet', '-DDAZZLE_COMPILATION', '-lssp' ],
   )
 
   if get_option('with_vapi')
