Patch for building on Solaris.  Evaluate for sending upstream

diff --git a/sys/v4l2/gstv4l2.c b/sys/v4l2/gstv4l2.c
index cd75ec5..e2e5ca1 100644
--- a/sys/v4l2/gstv4l2.c
+++ b/sys/v4l2/gstv4l2.c
@@ -39,6 +39,10 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "ext/videodev2.h"
 #include "v4l2-utils.h"
 
diff --git a/sys/v4l2/gstv4l2allocator.c b/sys/v4l2/gstv4l2allocator.c
index e82a853..9491009 100644
--- a/sys/v4l2/gstv4l2allocator.c
+++ b/sys/v4l2/gstv4l2allocator.c
@@ -25,6 +25,10 @@
 # define _GNU_SOURCE            /* O_CLOEXEC */
 #endif
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "ext/videodev2.h"
 
 #include "gstv4l2object.h"
diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 18f019a..bf991b8 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -35,6 +35,10 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "gst/video/video.h"
 #include "gst/video/gstvideometa.h"
 #include "gst/video/gstvideopool.h"
diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 65a7ff0..44f7fa0 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -34,6 +34,10 @@
 #include <gudev/gudev.h>
 #endif
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "ext/videodev2.h"
 #include "gstv4l2object.h"
 #include "gstv4l2tuner.h"
diff --git a/sys/v4l2/gstv4l2radio.c b/sys/v4l2/gstv4l2radio.c
index ab7c76d..2f4b519 100644
--- a/sys/v4l2/gstv4l2radio.c
+++ b/sys/v4l2/gstv4l2radio.c
@@ -42,6 +42,10 @@
 
 #include <string.h>
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "gst/gst-i18n-plugin.h"
 
 #include "gstv4l2object.h"
diff --git a/sys/v4l2/gstv4l2sink.c b/sys/v4l2/gstv4l2sink.c
index a971ef4..9d4e3dd 100644
--- a/sys/v4l2/gstv4l2sink.c
+++ b/sys/v4l2/gstv4l2sink.c
@@ -53,6 +53,10 @@
 #include <config.h>
 #endif
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 #include "gst/video/gstvideometa.h"
 
 #include "gstv4l2colorbalance.h"
diff --git a/sys/v4l2/gstv4l2videodec.c b/sys/v4l2/gstv4l2videodec.c
index 2eb0757..44d6dac 100644
--- a/sys/v4l2/gstv4l2videodec.c
+++ b/sys/v4l2/gstv4l2videodec.c
@@ -42,6 +42,10 @@
 #include <string.h>
 #include <gst/gst-i18n-plugin.h>
 
+#ifdef HAVE_FIONREAD_IN_SYS_FILIO
+#include <sys/filio.h>
+#endif
+
 GST_DEBUG_CATEGORY_STATIC (gst_v4l2_video_dec_debug);
 #define GST_CAT_DEFAULT gst_v4l2_video_dec_debug
 
