# From Alan Coopersmith
diff --git a/doc/Doxyfile.in b/doc/Doxyfile.in
index 5881ff9..102771f 100644
--- a/doc/Doxyfile.in
+++ b/doc/Doxyfile.in
@@ -132,7 +132,7 @@ FULL_PATH_NAMES        = YES
 # relative paths, which will be relative from the directory where doxygen is
 # started.
 
-STRIP_FROM_PATH        =
+STRIP_FROM_PATH        = @abs_top_srcdir@
 
 # The STRIP_FROM_INC_PATH tag can be used to strip a user-defined part of
 # the path mentioned in the documentation of a class, which tells
