Use default C stardard to make suricata build.

--- suricata-7.0.6/configure.ac
+++ suricata-7.0.6/configure.ac
@@ -297,23 +297,6 @@ ;;
     esac
     AC_MSG_RESULT(ok)
 
-    # check if our target supports c11
-    AC_MSG_CHECKING(for c11 support)
-    OCFLAGS=$CFLAGS
-    CFLAGS="-std=c11"
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdlib.h>]],
-                [[ static _Thread_local int i; i = 1; i++; ]])],
-            AC_MSG_RESULT([yes])
-            [AC_DEFINE([TLS_C11], [1], [C11 Thread local storage])
-             CFLAGS="$OCFLAGS -std=c11"],
-            [AC_MSG_RESULT([no])
-             CFLAGS="$OCFLAGS"
-             have_c11=no
-             have_c11_tls=no])
-    if [ test "x$have_c11" = "xno" ]; then
-        CFLAGS="$CFLAGS -std=gnu99"
-    fi
-
     # check if our target supports thread local storage
     AC_MSG_CHECKING(for thread local storage gnu __thread support)
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdlib.h>]],
