Use Solaris existing network devices name.

Not for upstream.

--- a/suricata.yaml.in
+++ b/suricata.yaml.in
@@ -812,7 +812,7 @@ copy-iface: none
 
 # Cross platform libpcap capture support
 pcap:
-  - interface: eth0
+  - interface: net0
     # On Linux, pcap will try to use mmap'ed capture and will use "buffer-size"
     # as total memory used by the ring. So set this to something bigger
     # than 1% of your bandwidth.
