Solaris syslog.h doesn't define LOG_FTP.

--- a/src/util-syslog.c
+++ b/src/util-syslog.c
@@ -33,7 +33,9 @@
     { "authpriv",       LOG_AUTHPRIV },
     { "cron",           LOG_CRON },
     { "daemon",         LOG_DAEMON },
+#if defined(LOG_FTP)
     { "ftp",            LOG_FTP },
+#endif
     { "kern",           LOG_KERN },
     { "lpr",            LOG_LPR },
     { "mail",           LOG_MAIL },
