https://github.com/vim/vim/issues/13084


--- vim-9.1.0652/src/auto/configure
+++ vim-9.1.0652/src/auto/configure
@@ -6592,7 +6592,7 @@ if test "$perl_ok" = "yes" -a "X$libperl
       printf "%s\n" "#define DYNAMIC_PERL 1" >>confdefs.h
 
       PERL_CFLAGS="-DDYNAMIC_PERL_DLL=\\\"$libperl\\\" $PERL_CFLAGS"
-      PERL_LIBS=""
+      PERL_LIBS="`echo $PERL_LIBS | sed s/-lperl//`"
     fi
   fi
 
--- vim-9.1.0652/src/configure.ac
+++ vim-9.1.0652/src/configure.ac
@@ -1252,7 +1252,7 @@ if test "$enable_perlinterp" = "dynamic"
     if test "$perl_ok" = "yes" -a "X$libperl" != "X"; then
       AC_DEFINE(DYNAMIC_PERL)
       PERL_CFLAGS="-DDYNAMIC_PERL_DLL=\\\"$libperl\\\" $PERL_CFLAGS"
-      PERL_LIBS=""
+      PERL_LIBS="`echo $PERL_LIBS | sed s/-lperl//`"
     fi
   fi
 
