pytest-cov is not available on Solaris.

--- wcwidth-0.2.13/tox.ini
+++ wcwidth-0.2.13/tox.ini
@@ -38,8 +38,7 @@ atomic = true
 [pytest]
 norecursedirs = .git .tox build
 addopts = --disable-pytest-warnings
-          --cov-append --cov-report=html --color=yes --ignore=setup.py --ignore=.tox
-          --cov=wcwidth
+          --color=yes --ignore=setup.py --ignore=.tox
 filterwarnings =
     error
 junit_family = xunit1
