Temporary disable warning to allow smooth ON transition (35633884).

--- binutils-2.40/gas/config/tc-i386.c
+++ binutils-2.40/gas/config/tc-i386.c
@@ -7422,13 +7422,6 @@ as_bad (_("no instruction mnemonic suffi
 			"no register operands; can't size `%s'"), i.tm.name);
 	      return 0;
 	    }
-	  if (operand_check == check_warning)
-	    as_warn (_("%s; using default for `%s'"),
-		       intel_syntax
-		       ? _("ambiguous operand size")
-		       : _("no instruction mnemonic suffix given and "
-			   "no register operands"),
-		       i.tm.name);
 
 	  if (i.tm.opcode_modifier.floatmf)
 	    i.suffix = SHORT_MNEM_SUFFIX;
