Solaris specific patch. Describes a change in OpenSSL defaults.

--- fetchmail-6.4.22/fetchmail.man.orig	2021-09-17 16:56:20.384305916 +0200
+++ fetchmail-6.4.22/fetchmail.man	2021-09-17 16:58:42.296632432 +0200
@@ -522,9 +522,9 @@
 supported by the particular installed OpenSSL library.
 .RS
 .IP '\fBTLS1.2+\fP'
-(recommended). Since v6.4.0. Require TLS. Auto-negotiate TLSv1.2 or newer.
+(default in Oracle Solaris since v6.4.22). Since v6.4.0. Require TLS. Auto-negotiate TLSv1.2 or newer.
 .IP '\fBauto\fP'
-(default). Since v6.4.0. Require TLS. Auto-negotiate TLSv1 or newer, disable 
+. Since v6.4.0. Require TLS. Auto-negotiate TLSv1 or newer, disable 
 SSLv3 downgrade.
 (fetchmail 6.3.26 and older have auto-negotiated all protocols that
 their OpenSSL library supported, including the broken SSLv3).
