For 32/64 tests, pick up the appropriate pkg_config_path

--- a/tests/Makefile.am	2023-10-10 12:14:56.552122740 -0700
+++ b/tests/Makefile.am	2023-10-10 12:15:15.029133881 -0700
@@ -32,6 +32,7 @@
 	export CFLAGS='$(CFLAGS)'; \
 	export CPPFLAGS='$(CPPFLAGS)'; \
 	export LDFLAGS='$(LDFLAGS)'; \
+	export PKG_CONFIG_PATH='$(PKG_CONFIG_PATH)'; \
 	export TEST_CFLAGS='$(TEST_CFLAGS)'; \
 	export abs_top_srcdir='$(abs_top_srcdir)'; \
 	export abs_top_builddir='$(abs_top_builddir)'; \
