Solaris doesn't have MAP_FILE

Not suitable for upstream

--- webkitgtk-2.30.3/Source/WTF/wtf/FileSystem.cpp	2020-11-23 10:42:56.384528155 +0000
+++ webkitgtk-2.30.3/Source/WTF/wtf/FileSystem.cpp	2020-11-23 10:42:18.923396550 +0000
@@ -339,7 +339,7 @@ bool MappedFileData::mapFileHandle(Platf
 #endif
     }
 
-    void* data = mmap(0, size, pageProtection, MAP_FILE | (mapMode == MappedFileMode::Shared ? MAP_SHARED : MAP_PRIVATE), fd, 0);
+    void* data = mmap(0, size, pageProtection, (mapMode == MappedFileMode::Shared ? MAP_SHARED : MAP_PRIVATE), fd, 0);
 
     if (data == MAP_FAILED) {
         return false;
