Solaris doesn't have MAP_FILE

Not suitable for upstream

--- a/Source/WebCore/platform/FileSystem.cpp	2018-02-18 18:59:38.879497608 -0800
+++ b/Source/WebCore/platform/FileSystem.cpp	2018-02-18 19:00:03.699372671 -0800
@@ -316,7 +316,7 @@
         return;
     }
 
-    void* data = mmap(0, size, PROT_READ, MAP_FILE | MAP_SHARED, fd, 0);
+    void* data = mmap(0, size, PROT_READ, MAP_SHARED, fd, 0);
     close(fd);
 
     if (data == MAP_FAILED) {
