Disable for Solaris

Not suitable for upstream

diff --git a/Source/WTF/wtf/PlatformUse.h b/Source/WTF/wtf/PlatformUse.h
index ade42dd..c8e94fa 100644
--- a/Source/WTF/wtf/PlatformUse.h
+++ b/Source/WTF/wtf/PlatformUse.h
@@ -92,8 +92,10 @@
 #endif
 
 #if PLATFORM(COCOA) || (PLATFORM(GTK) || PLATFORM(WPE))
+#if !OS(SOLARIS)
 #define USE_FILE_LOCK 1
 #endif
+#endif
 
 #if PLATFORM(COCOA)
 #define USE_FOUNDATION 1
