Disable for Solaris

Not suitable for upstream

--- a/Source/WTF/wtf/Platform.h	2018-02-23 15:00:14.639341856 -0800
+++ b/Source/WTF/wtf/Platform.h	2018-02-23 15:00:54.557763796 -0800
@@ -550,8 +550,10 @@
 #define USE_HARFBUZZ 1
 #define USE_SOUP 1
 #define USE_WEBP 1
+#if !OS(SOLARIS)
 #define USE_FILE_LOCK 1
 #endif
+#endif
 
 #if PLATFORM(GTK)
 #define GLIB_VERSION_MIN_REQUIRED GLIB_VERSION_2_36
