Source: Desktop consolidation
Patch Comment: type:branding bugid:263 bugster:6794539

--- a/avahi-daemon/avahi-dbus.conf.in
+++ b/avahi-daemon/avahi-dbus.conf.in
@@ -22,11 +22,6 @@ <deny send_destination="org.freedesktop.
           send_interface="org.freedesktop.Avahi.Server2" send_member="SetHostName"/>
   </policy>
 
-  <!-- Allow everything, including access to SetHostName to users of the group "@AVAHI_PRIV_ACCESS_GROUP@" -->
-  <policy group="@AVAHI_PRIV_ACCESS_GROUP@">
-    <allow send_destination="org.freedesktop.Avahi"/>
-    <allow receive_sender="org.freedesktop.Avahi"/>
-  </policy>
   <policy user="root">
     <allow send_destination="org.freedesktop.Avahi"/>
     <allow receive_sender="org.freedesktop.Avahi"/>
