From c5641702b552d71bc99c1ba3b45073488a30d398 Mon Sep 17 00:00:00 2001
From: Adam Turner <9087854+aa-turner@users.noreply.github.com>
Date: Sun, 5 Mar 2023 13:32:37 +0000
Subject: [PATCH] Update tests for Babel 2.12 (CLDR 42)

---
 tests/test_util_i18n.py | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/tests/test_util_i18n.py b/tests/test_util_i18n.py
index e52e27671d8..7be6f3e779a 100644
--- a/tests/test_util_i18n.py
+++ b/tests/test_util_i18n.py
@@ -4,12 +4,15 @@
 import os
 import warnings
 
+import babel
 import pytest
 from babel.messages.mofile import read_mo
 
 from sphinx.errors import SphinxError
 from sphinx.util import i18n
 
+BABEL_VERSION = tuple(map(int, babel.__version__.split('.')))
+
 
 def test_catalog_info_for_file_and_path():
     cat = i18n.CatalogInfo('path', 'domain', 'utf-8')
@@ -78,10 +81,16 @@ def test_format_date():
     format = '%x'
     assert i18n.format_date(format, date=datet, language='en') == 'Feb 7, 2016'
     format = '%X'
-    assert i18n.format_date(format, date=datet, language='en') == '5:11:17 AM'
+    if BABEL_VERSION >= (2, 12):
+        assert i18n.format_date(format, date=datet, language='en') == '5:11:17\u202fAM'
+    else:
+        assert i18n.format_date(format, date=datet, language='en') == '5:11:17 AM'
     assert i18n.format_date(format, date=date, language='en') == 'Feb 7, 2016'
     format = '%c'
-    assert i18n.format_date(format, date=datet, language='en') == 'Feb 7, 2016, 5:11:17 AM'
+    if BABEL_VERSION >= (2, 12):
+        assert i18n.format_date(format, date=datet, language='en') == 'Feb 7, 2016, 5:11:17\u202fAM'
+    else:
+        assert i18n.format_date(format, date=datet, language='en') == 'Feb 7, 2016, 5:11:17 AM'
     assert i18n.format_date(format, date=date, language='en') == 'Feb 7, 2016'
 
     # timezone
