https://bugzilla.mozilla.org/show_bug.cgi?id=1391072

diff -r 7455c74d833a build/autoconf/config.guess
--- a/build/autoconf/config.guess	Thu Jun 22 20:44:50 2017 -0700
+++ b/build/autoconf/config.guess	Mon Jun 26 10:30:03 2017 +0000
@@ -373,7 +373,7 @@
 	echo sparc-hal-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
     sun4*:SunOS:5.*:* | tadpole*:SunOS:5.*:*)
-	echo sparc-sun-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
+	echo sparc64-sun-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
     i86pc:AuroraUX:5.*:* | i86xen:AuroraUX:5.*:*)
 	echo i386-pc-auroraux${UNAME_RELEASE}
