https://bugzilla.mozilla.org/show_bug.cgi?id=1144632

--- firefox-60.1.0/other-licenses/snappy/moz.build
+++ firefox-60.1.0/other-licenses/snappy/moz.build
@@ -28,3 +28,6 @@
         '-Wno-sign-compare',
         '-Wno-unused-function'
     ]
+
+if CONFIG['TARGET_ENDIANNESS'] == 'big':
+    DEFINES['IS_BIG_ENDIAN'] = 1
--- firefox-60.1.0/build/moz.configure/init.configure
+++ firefox-60.1.0/build/moz.configure/init.configure
@@ -848,6 +848,7 @@
 set_config('INTEL_ARCHITECTURE', target_variables.INTEL_ARCHITECTURE)
 set_config('TARGET_CPU', target.raw_cpu)
 set_config('TARGET_OS', target.raw_os)
+set_config('TARGET_ENDIANNESS', target.endianness)
 
 
 @depends(host)
