Debug build install manifest issue. Should be investigated.

--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -195,7 +195,7 @@
 @RESPATH@/components/dom.xpt
 @RESPATH@/components/dom_base.xpt
 @RESPATH@/components/dom_bindings.xpt
-#ifdef MOZ_DEBUG
+#ifdef MOZ_DEBUGXXX
 @RESPATH@/components/dom_bindings_test.xpt
 #endif
 @RESPATH@/components/dom_file.xpt
@@ -240,7 +240,7 @@
 @RESPATH@/components/jsdebugger.xpt
 @RESPATH@/browser/components/jsinspector.xpt
 @RESPATH@/components/layout_base.xpt
-#ifdef MOZ_DEBUG
+#ifdef MOZ_DEBUGXXX
 @RESPATH@/components/layout_debug.xpt
 #endif
 #ifdef NS_PRINTING
