https://bugzilla.mozilla.org/show_bug.cgi?id=894240

--- mozjs-24.2.0/js/src/shell/jsoptparse.cpp
+++ mozjs-24.2.0/js/src/shell/jsoptparse.cpp
@@ -253,7 +253,7 @@
     char *eq = strchr(argv[*i], '=');
     if (eq) {
         *value = eq + 1;
-        if (value[0] == '\0')
+        if (*value[0] == '\0')
             return error("A value is required for option %.*s", eq - argv[*i], argv[*i]);
         return Okay;
     }
