No gudev support in Solaris.

Not suitable for upstream

--- a/meson.build	2021-02-16 11:33:48.074314845 -0800
+++ b/meson.build	2021-02-16 11:33:53.964487500 -0800
@@ -131,7 +131,6 @@
 pulse_dep = dependency('libpulse', version: pulse_req_version)
 pulse_mainloop_dep = dependency('libpulse-mainloop-glib', version: pulse_req_version)
 upower_glib_dep = dependency('upower-glib', version: '>= 0.99.8')
-gudev_dep = dependency('gudev-1.0', version: '>= 232')
 x11_dep = dependency('x11')
 xi_dep = dependency('xi', version: '>= 1.2')
 epoxy_dep = dependency('epoxy')
--- a/panels/common/meson.build	2021-02-16 11:34:12.314133080 -0800
+++ b/panels/common/meson.build	2021-02-16 11:34:54.626329445 -0800
@@ -85,18 +85,16 @@
 )
 
 gsd_headers = [
-  'gsd-device-manager.h',
   'gsd-input-helper.h'
 ]
 
 gsd_sources = [
-  'gsd-device-manager.c',
   'gsd-input-helper.c'
 ]
 
 sources = common_sources + files(gsd_sources)
 
-deps = common_deps + [ gudev_dep ]
+deps = common_deps 
 
 libdevice = static_library(
   'device',
--- a/panels/info-overview/meson.build	2021-02-16 11:35:15.576108675 -0800
+++ b/panels/info-overview/meson.build	2021-02-16 11:35:21.973799080 -0800
@@ -42,7 +42,6 @@
 )
 
 deps = common_deps + [
-  dependency('udisks2', version: '>= 2.8.2'),
   dependency('libgtop-2.0')
 ]
 
--- a/panels/meson.build	2021-02-16 11:35:33.762508500 -0800
+++ b/panels/meson.build	2021-02-16 11:35:40.850040105 -0800
@@ -14,7 +14,6 @@
   'location',
   'lock',
   'microphone',
-  'mouse',
   'notifications',
   'online-accounts',
   'region',
--- a/shell/cc-panel-loader.c	2021-02-16 11:35:57.919169960 -0800
+++ b/shell/cc-panel-loader.c	2021-02-16 11:36:13.810032735 -0800
@@ -42,7 +42,6 @@
 extern GType cc_display_panel_get_type (void);
 extern GType cc_info_overview_panel_get_type (void);
 extern GType cc_keyboard_panel_get_type (void);
-extern GType cc_mouse_panel_get_type (void);
 #ifdef BUILD_NETWORK
 extern GType cc_network_panel_get_type (void);
 extern GType cc_wifi_panel_get_type (void);
@@ -104,7 +103,6 @@
   PANEL_TYPE("location",         cc_location_panel_get_type,             NULL),
   PANEL_TYPE("lock",             cc_lock_panel_get_type,                 NULL),
   PANEL_TYPE("microphone",       cc_microphone_panel_get_type,           NULL),
-  PANEL_TYPE("mouse",            cc_mouse_panel_get_type,                NULL),
 #ifdef BUILD_NETWORK
   PANEL_TYPE("network",          cc_network_panel_get_type,              NULL),
   PANEL_TYPE("wifi",             cc_wifi_panel_get_type,                 cc_wifi_panel_static_init_func),
