disable the gnome-online-accounts registration page

--- a/panels/meson.build	2024-05-04 21:26:43.642987696 -0700
+++ b/panels/meson.build	2024-05-04 21:26:36.492872490 -0700
@@ -16,7 +16,6 @@
   'microphone',
   'multitasking',
   'notifications',
-  'online-accounts',
   'region',
   'removable-media',
   'search',
--- a/shell/cc-panel-loader.c	2024-05-04 21:35:48.638923809 -0700
+++ b/shell/cc-panel-loader.c	2024-05-04 21:36:07.176165680 -0700
@@ -116,7 +116,6 @@
   PANEL_TYPE("wifi",             cc_wifi_panel_get_type,                 cc_wifi_panel_static_init_func),
 #endif
   PANEL_TYPE("notifications",    cc_notifications_panel_get_type,        NULL),
-  PANEL_TYPE("online-accounts",  cc_goa_panel_get_type,                  NULL),
   PANEL_TYPE("region",           cc_region_panel_get_type,               NULL),
   PANEL_TYPE("removable-media",  cc_removable_media_panel_get_type,      NULL),
   PANEL_TYPE("search",           cc_search_panel_get_type,               NULL),
