From 817c3f71030ca9b4f88946ecf929bfaeaaad5ed1 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 10 Sep 2016 09:46:02 -0700
Subject: [PATCH:mesa] Can't use restrict in C++

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 include/c99_compat.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git unpacked-src/include/c99_compat.h unpacked-src/include/c99_compat.h
index 24e96e0..1e9e12b 100644
--- a/include/c99_compat.h
+++ b/include/c99_compat.h
@@ -96,7 +96,7 @@
  * - http://cellperformance.beyond3d.com/articles/2006/05/demystifying-the-restrict-keyword.html
  */
 #ifndef restrict
-#  if (__STDC_VERSION__ >= 199901L)
+#  if (__STDC_VERSION__ >= 199901L) && !defined(__cplusplus)
      /* C99 */
 #  elif defined(__GNUC__)
 #    define restrict __restrict__
-- 
2.7.4

