From 9054be01240cf9d9e8a92fc7289e2d861bf530d6 Mon Sep 17 00:00:00 2001
From: Jory Pratt <anarchy@gentoo.org>
Date: Thu, 22 Aug 2019 14:42:59 -0500
Subject: [PATCH] Add option to control elf-tls

Signed-off-by: Jory Pratt <anarchy@gentoo.org>
---
 meson.build       | 3 +--
 meson_options.txt | 6 ++++++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index bf61511..328ff66 100644
--- a/meson.build
+++ b/meson.build
@@ -392,7 +392,7 @@
 endif
 
 # Android uses emutls for versions <= P/28. For USE_ELF_TLS we need ELF TLS.
-if host_machine.system() != 'windows' and (not with_platform_android or get_option('platform-sdk-version') >= 29)
+if get_option('elf-tls')
   pre_args += '-DUSE_ELF_TLS'
 endif
 
diff --git a/meson_options.txt b/meson_options.txt
index b768c15..2c19e9e 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -285,6 +285,12 @@ option(
   value : false,
   description : 'Build an SELinux-aware Mesa'
 )
+option(
+  'elf-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Enable thread-local storage in GLX and EGL'
+)
 option(
   'osmesa',
   type : 'combo',
-- 
2.23.0

