--- a/lib/facter/resolvers/solaris/ffi/functions.rb.orig	2020-11-29 23:10:27.094298018 +0000
+++ b/lib/facter/resolvers/solaris/ffi/functions.rb	2020-11-29 23:10:42.408640581 +0000
@@ -6,7 +6,7 @@
       module FFI
         module Ioctl
           extend ::FFI::Library
-          ffi_lib ::FFI::Library::LIBC, '/usr/lib/libsocket.so'
+          ffi_lib ::FFI::Library::LIBC, '/usr/lib/64/libsocket.so'
 
           attach_function :ioctl_base, :ioctl, %i[int int pointer], :int
           attach_function :open_socket, :socket, %i[int int int], :int
