Add SunOS to list or supported OSs.

--- zstd-1.5.1/Makefile.orig	2022-02-01 15:17:30.941889470 +0100
+++ zstd-1.5.1/Makefile	2022-02-01 15:18:05.497482395 +0100
@@ -149,7 +149,7 @@
 #------------------------------------------------------------------------------
 # make install is validated only for Linux, macOS, Hurd and some BSD targets
 #------------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT Haiku AIX))
+ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT Haiku AIX SunOS))
 
 HOST_OS = POSIX
 
