Add SunOS to list or supported OSs.

--- zstd-1.5.0/Makefile.orig	2021-10-15 11:25:36.918662396 +0200
+++ zstd-1.5.0/Makefile	2021-10-15 11:26:03.889941742 +0200
@@ -148,7 +148,7 @@
 #------------------------------------------------------------------------------
 # make install is validated only for Linux, macOS, Hurd and some BSD targets
 #------------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT Haiku))
+ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD DragonFly NetBSD MSYS_NT Haiku SunOS))
 
 HOST_OS = POSIX
 
