--- libidn-1.44/java/Makefile.in
+++ libidn-1.44/java/Makefile.in
@@ -1698,7 +1698,7 @@ top_srcdir = @top_srcdir@
 SUBDIRS = src
 EXTRA_DIST = LICENSE-2.0.txt
 INSTALL_TARGETS = $(am__append_1)
-@JAVA_TRUE@libidn_jardir = $(datadir)/java
+@JAVA_TRUE@libidn_jardir = $(datadir)/lib/idn
 @JAVA_TRUE@libidn_jar_DATA = libidn-$(VERSION).jar
 all: all-recursive
 
@@ -2078,7 +2078,7 @@ @JAVA_TRUE@			gnu/inet/encoding/Stringpr
 @JAVA_TRUE@		-C $(builddir)/src/main/java \
 @JAVA_TRUE@			gnu/inet/encoding/StringprepException.class
 @JAVA_TRUE@install-jar:
-@JAVA_TRUE@	install -d -m 755 $(DESTDIR)$(libidn_jardir)
+@JAVA_TRUE@	$(INSTALL) -d -m 755 $(DESTDIR)$(libidn_jardir)
 @JAVA_TRUE@	ln -sf libidn-$(VERSION).jar $(DESTDIR)$(libidn_jardir)/libidn.jar
 
 @JAVA_TRUE@clean-local:
