--- a/main.c.orig2	2016-06-03 00:19:39.914041433 -0700
+++ b/main.c	2016-07-07 23:44:11.858793092 -0700
@@ -2441,6 +2441,14 @@
 	    SysError(ERROR_LUMALLOC);
 
 	x_appendargv(command_to_exec_with_luit, split_filter);
+	
+	if (Console) {
+	    char *encoding_opt1[2];
+	    int pos = 0;
+	    encoding_opt1[pos++] = "-C";
+	    encoding_opt1[pos] = 0;
+	    x_appendargv(command_to_exec_with_luit, encoding_opt1);
+	}
 	if (count_using) {
 	    char *encoding_opt[4];
 	    encoding_opt[0] = x_strdup("-encoding");
