These changes have been sent upstream.

diff -urN sg3_utils-1.28/lib/Makefile.am sg3_utils-1.28.new/lib/Makefile.am
--- sg3_utils-1.28/lib/Makefile.am	2011-03-08 15:16:10.684395224 +0800
+++ sg3_utils-1.28.new/lib/Makefile.am	2011-03-08 15:16:52.811149215 +0800
@@ -41,8 +41,8 @@
 endif

 # -std=<s> can be c99, c11, gnu11, etc. Default is gnu89 (gnu90 is the same)
 # -Wall is no longer all warnings. Add -W (since renamed to -Wextra) for more
-AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-AM_CFLAGS = -Wall -W
+/*AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+AM_CFLAGS = -v
 # AM_CFLAGS = -Wall -W -pedantic -std=c11
 # AM_CFLAGS = -Wall -W -pedantic -std=c++11
 
 
--- sg3_utils-1.28/lib/Makefile.in	2015-12-01 13:03:58.000000000 +0000
+++ sg3_utils-1.28.new/lib/Makefile.in  2017-04-18 14:49:43.833137845 +0000
@@ -349,8 +349,8 @@

 # -std=<s> can be c99, c11, gnu11, etc. Default is gnu89 (gnu90 is the same)
 # -Wall is no longer all warnings. Add -W (since renamed to -Wextra) for more
-AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-AM_CFLAGS = -Wall -W
+/*AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+AM_CFLAGS = -v
 # AM_CFLAGS = -Wall -W -pedantic -std=c11
 # AM_CFLAGS = -Wall -W -pedantic -std=c++11
 lib_LTLIBRARIES = libsgutils2.la


--- sg3_utils-1.28/src/Makefile.am 	2016-02-07 20:53:45.000000000 +0000
+++ sg3_utils-1.28.new/src/Makefile.am  2017-04-18 16:32:42.989663173 +0000
@@ -41,8 +41,8 @@

 # -std=<s> can be c99, c11, gnu11, etc. Default is gnu89 (gnu90 is the same)
 # -Wall is no longer all warnings. Add -W (since renamed to -Wextra) for more
-AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-AM_CFLAGS = -Wall -W @os_cflags@
+/*AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+AM_CFLAGS = -v @os_cflags@
 # AM_CFLAGS = -Wall -W @os_cflags@ -pedantic -std=c11
 # AM_CFLAGS = -Wall -W @os_cflags@ -pedantic -std=c++11


--- sg3_utils-1.42/src/Makefile.in       2016-02-07 20:53:45.000000000 +0000
+++ sg3_utils-1.42.new/src/Makefile.in   2017-04-18 16:11:55.759736785 +0000
@@ -557,8 +557,8 @@

 # -std=<s> can be c99, c11, gnu11, etc. Default is gnu89 (gnu90 is the same)
 # -Wall is no longer all warnings. Add -W (since renamed to -Wextra) for more
-AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-AM_CFLAGS = -Wall -W @os_cflags@
+/*AM_CPPFLAGS = -iquote ${top_srcdir}/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+AM_CFLAGS = -v @os_cflags@
 # AM_CFLAGS = -Wall -W @os_cflags@ -pedantic -std=c11
 # AM_CFLAGS = -Wall -W @os_cflags@ -pedantic -std=c++11
 sg_compare_and_write_LDADD = ../lib/libsgutils2.la @os_libs@


--- sg3_utils-1.42/src/sg_raw.c          2015-12-20 08:23:44.000000000 +0000
+++ sg3_utils-1.42.new/src/sg_raw.c      2017-04-19 11:31:32.899368686 +0000
@@ -12,7 +12,7 @@
  * data phase) through a Generic SCSI interface.
  */

-#define _XOPEN_SOURCE 600       /* clear up posix_memalign() warning */
+/*#define _XOPEN_SOURCE 600       /* clear up posix_memalign() warning */

 #include <stdlib.h>
 #include <unistd.h>

