Override Solaris-specific settings => not for upstream.

diff --git a/PerlMagick/Makefile.PL.in b/PerlMagick/Makefile.PL.in
index 3256247..0d1f464 100644
--- a/PerlMagick/Makefile.PL.in
+++ b/PerlMagick/Makefile.PL.in
@@ -214,6 +214,8 @@ WriteMakefile
    # C compiler flags (e.g. -O -g)
    'CCFLAGS' => $CCFLAGS_magick,
 
+   'CCCDLFLAGS' => '-fPIC',
+
    # Linker
    #'LD' => $Config{'ld'} == $Config{'cc'} ? '@CC@' : $Config{'ld'},
 
@@ -232,8 +234,7 @@ WriteMakefile
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
 
-   # Let CFLAGS drive optimization flags by setting OPTIMIZE to empty
-   # 'OPTIMIZE'	=> '',
+   'OPTIMIZE'	=> '-O3',
 
    # Use same compiler as ImageMagick
    'PERLMAINCC'	=> '@PERLMAINCC@ @OPENMP_CFLAGS@',
diff --git a/PerlMagick/quantum/Makefile.PL.in b/PerlMagick/quantum/Makefile.PL.in
index 1778082..ce8a765 100644
--- a/PerlMagick/quantum/Makefile.PL.in
+++ b/PerlMagick/quantum/Makefile.PL.in
@@ -211,6 +211,8 @@ WriteMakefile
    # C compiler flags (e.g. -O -g)
    'CCFLAGS' => $CCFLAGS_magick,
 
+   'CCCDLFLAGS' => '-fPIC',
+
    # Linker
    #'LD' => $Config{'ld'} == $Config{'cc'} ? '@CC@' : $Config{'ld'},
 
@@ -229,8 +231,7 @@ WriteMakefile
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
 
-   # Let CFLAGS drive optimization flags by setting OPTIMIZE to empty
-   # 'OPTIMIZE'	=> '',
+   'OPTIMIZE'	=> '-O3',
 
    # Use same compiler as ImageMagick
    'PERLMAINCC'	=> '@PERLMAINCC@ @OPENMP_CFLAGS@',
