Existing set of patches from the X gate

diff --git a/man/scanpci.man b/man/scanpci.man
index a748edf..97336f8 100644
--- a/man/scanpci.man
+++ b/man/scanpci.man
@@ -35,10 +35,16 @@ about the configuration space settings for each PCI device.
 On most platforms,
 .I scanpci
 can only be run by the root user.
+On Oracle Solaris it may also be run using the 
+.I "Desktop Configuration"
+rights profile.  See \fBrbac\fR(__miscmansuffix__) and \fBuser_attr\fR(__filemansuffix__).
 .SH OPTIONS
 .TP 8
 .B \-v
 Print the configuration space information for each device in a verbose
 format.  Without this option, only a brief description is printed for
 each device.
+.SH "SEE ALSO"
+.BR prtconf "(__adminmansuffix__), " prtdiag "(__adminmansuffix__), " smbios "(__adminmansuffix__), "
+.BR user_attr "(__filemansuffix__), " rbac "(__miscmansuffix__)."
 
