From a4e8d675b4e0eeba79d4579bca8621bea7ee68fe Mon Sep 17 00:00:00 2001
From: Andrew Murray <radarhere@users.noreply.github.com>
Date: Wed, 18 Jun 2025 21:59:31 +1000
Subject: [PATCH] Only check DHT marker for libjpeg-turbo

---
 Tests/test_file_jpeg.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/Tests/test_file_jpeg.py b/Tests/test_file_jpeg.py
index 2827937cf57..614044d97eb 100644
--- a/Tests/test_file_jpeg.py
+++ b/Tests/test_file_jpeg.py
@@ -1064,10 +1064,16 @@ class TestFileJpeg:
         for marker in b"\xff\xd8", b"\xff\xd9":
             assert marker in data[1]
             assert marker in data[2]
-        # DHT, DQT
-        for marker in b"\xff\xc4", b"\xff\xdb":
+
+        # DQT
+        markers = [b"\xff\xdb"]
+        if features.check_feature("libjpeg_turbo"):
+            # DHT
+            markers.append(b"\xff\xc4")
+        for marker in markers:
             assert marker in data[1]
             assert marker not in data[2]
+
         # SOF0, SOS, APP0 (JFIF header)
         for marker in b"\xff\xc0", b"\xff\xda", b"\xff\xe0":
             assert marker not in data[1]
