diff -ur libsigsegv-2.10.orig/src/fault-solaris-i386.h libsigsegv-2.10/src/fault-solaris-i386.h
--- libsigsegv-2.10.orig/src/fault-solaris-i386.h	2020-02-18 12:54:26.107319750 +0000
+++ libsigsegv-2.10/src/fault-solaris-i386.h	2020-02-18 12:57:56.776341805 +0000
@@ -17,4 +17,4 @@
 
 #include "fault-solaris.h"
 
-#define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext.gregs[ESP]
+#define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext.gregs[REG_SP]
