Upstream contribution is planned but has not yet occurred.

--- postfix-3.2.2/makedefs.~1~	2017-06-15 12:28:28.279255178 +0000
+++ postfix-3.2.2/makedefs	2017-06-15 12:37:03.840492279 +0000
@@ -389,13 +389,16 @@
 		esac
 		# NISPLUS was removed after Solaris 10.
 		case $RELEASE in
-		    5.[0-9][0-9]*) CCARGS="$CCARGS -DNO_NISPLUS";;
+		    5.1[1-9]*) CCARGS="$CCARGS -DNO_NISPLUS";;
+		esac
+		# Work around broken str*casecmp(), fixed in Solaris 9. Do
+		# it all here instead of having half the solution in the
+		# sys_defs.h file.
+		case $RELEASE in
+		    5.[0-8]|5.[0-8].*) CCARGS="$CCARGS -Dstrcasecmp=fix_strcasecmp -Dstrncasecmp=fix_strncasecmp"
+		                       STRCASE="strcasecmp.o"
+		                       ;;
 		esac
-		# Work around broken str*casecmp(). Do it all here instead
-		# of having half the solution in the sys_defs.h file.
-		CCARGS="$CCARGS -Dstrcasecmp=fix_strcasecmp \
-		    -Dstrncasecmp=fix_strncasecmp"
-		STRCASE="strcasecmp.o"
 		# Avoid common types of braindamage
 		case "$LD_LIBRARY_PATH" in
 		?*) error "Don't set LD_LIBRARY_PATH";;
