Fix compiler warnings.

--- a/cogl/cogl/cogl-bitmap-conversion.c	2019-11-10 19:27:55.396167938 -0800
+++ b/cogl/cogl/cogl-bitmap-conversion.c	2019-11-10 19:28:22.651697292 -0800
@@ -133,7 +133,7 @@
    it is available. The same assembler code works for x86 and x86-64
    because it doesn't refer to any non-SSE registers directly */
 #if defined(__SSE2__) && defined(__GNUC__) \
-  && (defined(__x86_64) || defined(__i386))
+  && (defined(__x86_64) || defined(__i386)) && !defined(__sun)
 #define COGL_USE_PREMULT_SSE2
 #endif

--- a/src/backends/meta-monitor-manager-dummy.c 2019-11-12 10:09:00.387114245 -0800
+++ b/src/backends/meta-monitor-manager-dummy.c 2019-11-12 10:08:49.249512155 -0800
@@ -438,7 +438,7 @@
         }
     }

-  monitor_scales = g_newa (typeof (*monitor_scales), num_monitors);
+  monitor_scales = calloc (typeof (*monitor_scales), num_monitors);
   for (i = 0; i < num_monitors; i++)
     monitor_scales[i] = 1.0;


