--- libpcap-1.10.4/CMakeLists.txt
+++ libpcap-1.10.4/CMakeLists.txt
@@ -2747,7 +2747,7 @@ # is thread-safe (it wasn't thread-safe
     #
     add_definitions(-D_TS_ERRNO)
 
-    if(CMAKE_SYSTEM_VERSION STREQUAL "5.12")
+    if(EXISTS /usr/share/man/man7/attributes.7)
     else()
         #
         # Use System V conventions for man pages.
--- libpcap-1.10.4/configure.ac
+++ libpcap-1.10.4/configure.ac
@@ -2474,12 +2474,7 @@ # is thread-safe (it wasn't thread-safe
 	#
 	V_CCOPT="$V_CCOPT -D_TS_ERRNO"
 
-	case "`uname -r`" in
-
-	5.12)
-		;;
-
-	*)
+	if ! test -f /usr/share/man/man7/attributes.7 ; then
 		#
 		# Use System V conventions for man pages.
 		#
@@ -2487,8 +2482,7 @@ MAN_ADMIN_COMMANDS=1m
 		MAN_FILE_FORMATS=4
 		MAN_MISC_INFO=5
 		MAN_DEVICES=7D
-	esac
-	;;
+	fi
 esac
 AC_SUBST(V_LIB_CCOPT_FAT)
 AC_SUBST(V_LIB_LDFLAGS_FAT)
