From 0ed73181eb54e3f2dc744271c0329620972a85ab Mon Sep 17 00:00:00 2001
From: "Charles R. Anderson" <cra@wpi.edu>
Date: Sun, 2 Feb 2020 00:42:37 -0500
Subject: [PATCH] GCC 10 compatibility: use extern in header files when
 declaring global variables

---
 src/fping.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fping.h b/src/fping.h
index 44be999..232e7ea 100644
--- a/src/fping.h
+++ b/src/fping.h
@@ -11,7 +11,7 @@
 void crash_and_burn( char *message );
 void errno_crash_and_burn( char *message );
 int in_cksum( unsigned short *p, int n );
-int random_data_flag;
+extern int random_data_flag;
 
 /* socket.c */
 int  open_ping_socket_ipv4();
