--- ppdc/Makefile	2018-12-06 09:27:31.454495891 +0100
+++ ppdc/Makefile	2018-12-06 09:27:17.478683377 +0100
@@ -19,6 +19,9 @@
 
 include ../Makedefs
 
+# Make library self-contained.
+LINKCUPS += -lCrun -lm
+
 #
 # Object files...
 #

--- config-scripts/cups-threads.m4	2009-02-10 18:05:35.000000000 +0100
+++ config-scripts/cups-threads.m4	2018-12-06 10:25:59.992494239 +0100
@@ -40,6 +40,8 @@
 				# be POSIX-compliant... :(
 				if test $uname = SunOS; then
 					PTHREAD_FLAGS="$PTHREAD_FLAGS -D_POSIX_PTHREAD_SEMANTICS"
+					# No need to pass -lpthread since Solaris 11/11.
+					LIBS="$SAVELIBS"
 				fi
 				break
 			fi
