*** zebra/kernel_socket.c
--- zebra/kernel_socket.c
@@ -41,6 +41,12 @@
 extern struct zebra_privs_t zserv_privs;
 extern struct zebra_t zebrad;
 
+#if defined(SUNOS_5)
+/* Solaris hds struct sockaddr_in[6] definitions at 16 / 32 bytes size,
+ * so the whole concept doesn't really apply.
+ */
+#define ROUNDUP(a)      (a)
+# else
 /*
  * Given a sockaddr length, round it up to include pad bytes following
  * it.  Assumes the kernel pads to sizeof(long).
@@ -51,6 +57,7 @@
  */
 #define ROUNDUP(a) \
   ((a) > 0 ? (1 + (((a) - 1) | (sizeof(long) - 1))) : sizeof(long))
+#endif
 
 /*
  * Given a pointer (sockaddr or void *), return the number of bytes
