
--- a/testsuite/gtk/meson.build	2023-10-12 22:24:17.148377787 -0700
+++ b/testsuite/gtk/meson.build	2023-10-12 22:24:37.202352108 -0700
@@ -1,9 +1,9 @@
-gtk_tests_export_dynamic_ldflag = []
-
 if cc.get_id() != 'msvc' and cc.get_id() != 'clang'
   gtk_tests_export_dynamic_ldflag = ['-Wl,--export-dynamic']
 endif
 
+gtk_tests_export_dynamic_ldflag = []
+
 tests = [
   ['accel'],
   ['accessible'],
