Existing set of patches from the X gate

--- a/libkms/linux.c	Thu Apr 28 15:44:05 2016
+++ b/libkms/linux.c	Thu Apr 28 15:44:29 2016
@@ -40,6 +40,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #ifdef HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
--- a/xf86drm.h	Thu Apr 28 15:44:52 2016
+++ b/xf86drm.h	Thu Apr 28 15:45:29 2016
@@ -472,6 +472,18 @@
 #endif /* architecture */
 #endif /* __GNUC__ >= 2 */
 
+#if defined(__SUNPRO_C)
+#include <atomic.h>
+#define atomic_cmpset_int(p, c, n) ((c == atomic_cas_uint(p, c, n)) ? 1 : 0)
+#define DRM_CAS(lock,old,new,__ret)          \
+               do {                          \
+                                       unsigned int __result, __old = (old);\
+                                       __result = !atomic_cmpset_int(lock,__old,new);\
+                                       __ret = __result;          \
+                               } while(0)
+#endif
+
+
 #ifndef DRM_CAS
 #define DRM_CAS(lock,old,new,ret) do { ret=1; } while (0) /* FAST LOCK FAILS */
 #endif
--- a/xf86drmMode.c	Thu Apr 28 15:45:50 2016
+++ b/xf86drmMode.c	Thu Apr 28 15:46:23 2016
@@ -837,7 +837,9 @@
 	drmClose(fd);
 	return 0;
 #endif
-	return -ENOSYS;
+
+/* for now return 0 on solaris */
+       return 0;
 }
 
 int drmModeCrtcGetGamma(int fd, uint32_t crtc_id, uint32_t size,
--- a/xf86drm.c	Thu Apr 28 15:46:38 2016
+++ b/xf86drm.c	Thu Apr 28 15:46:56 2016
@@ -1105,7 +1105,7 @@
     drm_map_t map;
 
     memclear(map);
-    map.handle = (void *)(uintptr_t)handle;
+    map.handle = (drm_handle_t)(uintptr_t)handle;
 
     if(drmIoctl(fd, DRM_IOCTL_RM_MAP, &map))
 	return -errno;
