--- net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
@@ -311,7 +311,7 @@ if ((ksc = kstat_fd) == NULL) {
 	goto Return;
     }
     ks = kstat_lookup(ksc, NETSNMP_REMOVE_CONST(char *, statname),
-                      0, NETSNMP_REMOVE_CONST(char *, "link"));
+                      -1, NETSNMP_REMOVE_CONST(char *, "link"));
     if (ks != NULL) {
         kid = kstat_read(ksc, ks, NULL);
         if (kid == -1) {
