Solaris specific patch developed inhouse, not suitable for upstream
--- net-snmp-5.8/snmplib/scapi.c
+++ net-snmp-5.8/snmplib/scapi.c
@@ -205,8 +205,8 @@ };
 #ifdef NETSNMP_USE_OPENSSL
 static
 void EVP_hmac(const EVP_MD *evp_md, const void *key,
-              const unsigned char *msg, unsigned char *md,
-              size_t *md_len);
+              const unsigned char *msg, u_int msglen, 
+              unsigned char *md, size_t *md_len);
 static
 int EVP_decrypt(const EVP_CIPHER *type, u_char * key,
                 u_int keylen, u_char * iv, u_char * ciphertext,
@@ -794,7 +794,7 @@ }
 #ifndef NETSNMP_DISABLE_MD5
     if (auth_type == NETSNMP_USMAUTH_HMACMD5) {
         size_t buf_lens = buf_len;
-        EVP_hmac(hashfn, key, message, buf, &buf_lens);
+        EVP_hmac(hashfn, key, message, msglen, buf, &buf_lens);
         buf_len = (unsigned int) buf_lens;
     } else
 #endif
@@ -1833,8 +1833,8 @@ #endif /* NETSNMP_USE_INTERNAL_CRYPTO */
 #ifdef NETSNMP_USE_OPENSSL
 static
 void EVP_hmac(const EVP_MD *evp_md, const void *key,
-              const unsigned char *msg, unsigned char *md,
-              size_t *md_len)
+              const unsigned char *msg, u_int msglen, 
+              unsigned char *md, size_t *md_len)
 {
         EVP_MD_CTX* mdctx = NULL;
         EVP_PKEY *pkey = NULL;
@@ -1858,7 +1858,7 @@ EVP_MD_CTX_destroy(mdctx);
             EVP_PKEY_free(pkey);
             return;
         }
-        if(EVP_DigestSignUpdate(mdctx, msg, strlen(msg)) != 1) {
+        if(EVP_DigestSignUpdate(mdctx, msg, msglen) != 1) {
             snmp_log(LOG_ERR, "EVP_DigestSignUpdate failure.\n");
             EVP_MD_CTX_destroy(mdctx);
             EVP_PKEY_free(pkey);
