Solaris specific patch, not suitable for upstream.
--- net-snmp-5.9.5.2/man/snmpd.conf.5.def
+++ net-snmp-5.9.5.2/man/snmpd.conf.5.def
@@ -1868,7 +1868,7 @@ See the EXAMPLE.CONF file in the top-lev
 detailed example of how the above information is used in real
 examples.
 .SH "FILES"
-SYSCONFDIR/snmp/snmpd.conf
+DATADIR/snmp/snmpd.conf
 .SH "SEE ALSO"
 snmpconf(1), snmpusm(1), snmp.conf(5), snmp_config(5), snmpd(8), EXAMPLE.conf, netsnmp_config_api(3).
 .\" Local Variables:
--- net-snmp-5.9.5.2/man/snmptrapd.conf.5.def
+++ net-snmp-5.9.5.2/man/snmptrapd.conf.5.def
@@ -309,7 +309,7 @@ or any of \fI0\fR, \fIno\fR or \fIfalse\
 The default in each case is for the feature to be turned off, so these
 directives are typically only used to enable the appropriate behaviour.
 .SH FILES
-SYSCONFDIR/snmp/snmptrapd.conf
+DATADIR/snmp/snmptrapd.conf
 .SH "SEE ALSO"
 snmp_config(5), snmptrapd(8), syslog(8), traptoemail(1), variables(5), netsnmp_config_api(3).
 
