--- zsh-5.9/Completion/Solaris/Command/_zlogin.orig	2025-09-10 15:58:34.671390535 +0200
+++ zsh-5.9/Completion/Solaris/Command/_zlogin	2025-09-10 16:02:26.994240933 +0200
@@ -1,5 +1,5 @@
 #compdef zlogin
-# Synced with the Nevada build 162 man page
+# Synced with the S11U2 build 34 man page
 
 local -a _comp_priv_prefix
 _arguments -s \
@@ -6,6 +6,8 @@
   '-E[disable escape character]' \
   '-e[specify escape character]:character' \
   '-l[specify username]:username:_users' \
+  '-T[Enter immutable zone as a "Trusted Path Domain" member]' \
+  '-U[Like -T, but "unsafe"]' \
   ':zonename:_zones' \
   - set1 \
   '-C[connect to zone console]' \
