Fix path to zsh command.

--- zsh-5.9/Functions/VCS_Info/test-repo-git-rebase-apply.orig	2025-09-17 13:32:08.296456270 +0200
+++ zsh-5.9/Functions/VCS_Info/test-repo-git-rebase-apply	2025-09-17 13:32:46.797115144 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # This script creates a test repository for testing the git backend's behaviour during rebase-apply operations.
 #
--- zsh-5.9/Functions/VCS_Info/test-repo-git-rebase-merge.orig	2025-09-17 13:32:20.147894971 +0200
+++ zsh-5.9/Functions/VCS_Info/test-repo-git-rebase-merge	2025-09-17 13:33:04.448205407 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # This script creates a test repository for testing the git backend's behaviour during rebase-apply operations.
 #
