Changes to get a clean build on Solaris

--- a/src/libical/icalcomponent.c	2019-10-20 16:38:16.516992485 -0700
+++ b/src/libical/icalcomponent.c	2019-10-20 16:39:20.254854885 -0700
@@ -30,6 +30,7 @@
 
 #include <assert.h>
 #include <stdlib.h>
+#include <limits.h>
 
 struct icalcomponent_impl
 {
--- a/src/libical/icaltz-util.c	2019-10-20 16:38:09.205324545 -0700
+++ b/src/libical/icaltz-util.c	2019-10-20 16:38:36.634165620 -0700
@@ -26,6 +26,7 @@
 #include "icaltimezone.h"
 
 #include <stdlib.h>
+#include <limits.h>
 
 #if defined(sun) && defined(__SVR4)
 #include <sys/types.h>
